// @acutrack-bookprint/acutrack-ds/dist/docs/api/AlertDocumentation.ts
export default {
  title: 'Alert Component',
  description: 'Contextual feedback messages with multiple variants, sizes, and dismissible states for user notifications',
  
  // Basic usage examples
  basicUsage: `
    import { Alert } from '@acutrack-bookprint/acutrack-ds';
    
    // Basic alert
    <Alert variant="info" title="Information" content="This is an informational message." />
    
    // Success alert
    <Alert variant="success" title="Success!" content="Operation completed successfully." />
    
    // Warning alert
    <Alert variant="warning" title="Warning" content="Please review your input." />
    
    // Error alert
    <Alert variant="error" title="Error" content="Something went wrong." />
  `,
  
  // Advanced usage examples
  advancedUsage: `
    import { Alert, Icon } from '@acutrack-bookprint/acutrack-ds';
    
    // Dismissible alert
    <Alert 
      variant="info" 
      title="Dismissible Alert"
      content="This alert can be dismissed."
      isDismissible 
      onDismiss={() => setIsVisible(false)}
    />
    
    // With custom icon
    <Alert 
      variant="success"
      title="Custom Icon"
      content="This alert has a custom icon."
      customIcon={<Icon size="md">check_circle</Icon>}
    />
    
    // Different sizes
    <Alert variant="info" size="xs" title="Small Alert" />
    <Alert variant="info" size="lg" title="Large Alert" content="This is a larger alert." />
    
    // Without title
    <Alert variant="warning" content="This alert has no title." />
    
    // With children instead of content
    <Alert variant="info" title="Rich Content">
      <div>
        <p>This alert contains rich content.</p>
        <button>Action Button</button>
      </div>
    </Alert>
  `,
  
  // Available variants
  variants: {
    primary: 'Primary brand color alerts',
    secondary: 'Secondary information alerts',
    info: 'Informational content (blue theme)',
    success: 'Positive outcomes, completed states (green theme)',
    warning: 'Caution, attention needed (yellow theme)',
    error: 'Errors, failures, critical issues (red theme)'
  },
  
  // Available sizes
  sizes: {
    xs: 'Extra small alerts (12px text, minimal padding)',
    sm: 'Small alerts (14px text, compact padding)',
    md: 'Default size (16px text, standard padding)',
    lg: 'Large alerts (18px text, generous padding)',
    xl: 'Extra large alerts (20px text, maximum padding)'
  },
  
  // Key props
  keyProps: {
    variant: 'Alert variant (primary, secondary, info, success, warning, error)',
    size: 'Alert size (xs, sm, md, lg, xl) - affects icon size and padding',
    title: 'Alert title/heading text',
    content: 'Alert content/description text',
    children: 'Rich content instead of simple text content',
    isDismissible: 'Whether alert can be dismissed (shows close button)',
    onDismiss: 'Callback function when alert is dismissed',
    customIcon: 'Custom icon to override default variant icon',
    className: 'Additional CSS classes',
    style: 'Inline styles'
  },
  
  // Best practices
  bestPractices: {
    do: [
      'Use appropriate variants for different contexts (success for positive, warning for caution)',
      'Provide clear, concise titles and content',
      'Use dismissible alerts for temporary notifications',
      'Use custom icons when the default doesn\'t fit the context',
      'Keep alert content scannable and actionable',
      'Use appropriate sizes for the context (xs for inline, lg for prominent)'
    ],
    dont: [
      'Use too many alerts in a single view',
      'Make alert content too long or complex',
      'Use alerts for primary actions (use modals or pages instead)',
      'Mix different alert styles inconsistently',
      'Use alerts without considering accessibility',
      'Overuse dismissible alerts (they should be for temporary info)'
    ]
  },
  
  // Icon size mapping
  iconSizeMapping: {
    xs: 'sm',   // Extra small alerts use small icons
    sm: 'md',   // Small alerts use medium icons
    md: 'lg',   // Default alerts use large icons
    lg: 'xl',   // Large alerts use extra large icons
    xl: '2xl'   // Extra large alerts use 2xl icons
  },
  
  // Design system integration
  designSystem: {
    colors: 'Uses design system colors automatically (system-info, system-success, system-warning, system-error)',
    darkMode: 'Automatically supports dark mode with proper color variants',
    accessibility: 'Includes proper ARIA attributes, keyboard navigation, and screen reader support',
    consistency: 'Follows design system spacing, typography, and interaction patterns',
    icons: 'Automatically uses appropriate icons for each variant with proper sizing'
  },
  
  // Common use cases
  useCases: {
    formValidation: `
      <Alert variant="error" title="Validation Error" content="Please check your input." />
      <Alert variant="success" title="Success" content="Form submitted successfully." />
    `,
    systemNotifications: `
      <Alert variant="info" title="System Update" content="The system will be updated tonight." />
      <Alert variant="warning" title="Maintenance" content="Scheduled maintenance in 1 hour." />
    `,
    userFeedback: `
      <Alert variant="success" title="Profile Updated" content="Your profile has been saved." />
      <Alert variant="error" title="Save Failed" content="Unable to save changes. Please try again." />
    `,
    temporaryNotifications: `
      <Alert 
        variant="info" 
        title="New Feature" 
        content="Check out our new dashboard!"
        isDismissible 
        onDismiss={() => setShowFeatureAlert(false)}
      />
    `
  },
  
  // Accessibility features
  accessibility: {
    ariaRole: 'Uses role="alert" for important messages',
    keyboardNavigation: 'Dismissible alerts support keyboard interaction',
    screenReader: 'Proper ARIA labels and descriptions',
    focusManagement: 'Focus handling for dismissible alerts',
    colorContrast: 'Meets WCAG contrast requirements'
  },
  
  // Troubleshooting
  troubleshooting: {
    iconsNotScaling: [
      'Ensure you\'re using the size prop correctly',
      'Check that the Icon component is properly imported',
      'Verify that the icon size mapping is working'
    ],
    colorsNotAppearing: [
      'Ensure you\'re using design system variants',
      'Check that dark mode classes are properly applied',
      'Verify custom colors are valid CSS values'
    ],
    dismissibleNotWorking: [
      'Ensure isDismissible is set to true',
      'Provide an onDismiss handler function',
      'Check that the close button is not disabled'
    ],
    contentNotDisplaying: [
      'Ensure you\'re providing either content or children',
      'Check that the content is not empty or undefined',
      'Verify that the Alert component is properly imported'
    ]
  },
  
  // Size guidelines
  sizeGuidelines: {
    xs: 'Very small alerts, inline notifications (12px text)',
    sm: 'Small alerts, compact notifications (14px text)',
    md: 'Default size, most common use (16px text)',
    lg: 'Large alerts, prominent notifications (18px text)',
    xl: 'Extra large alerts, hero notifications (20px text)'
  },
  
  // Variant guidelines
  variantGuidelines: {
    primary: 'Main brand alerts, important announcements',
    secondary: 'Secondary information, less critical alerts',
    info: 'General information, tips, helpful content',
    success: 'Positive outcomes, completed actions, confirmations',
    warning: 'Caution, attention needed, potential issues',
    error: 'Errors, failures, critical issues, validation problems'
  }
};
