// @acutrack-bookprint/acutrack-ds/dist/docs/api/BadgeDocumentation.ts
export default {
  title: 'Badge Component',
  description: 'Flexible status indicators and labels with multiple variants, interactive states, and radius customization',
  version: '1.0.66',
  
  // Basic usage examples
  basicUsage: `
    import { Badge } from '@acutrack-bookprint/acutrack-ds';
    
    // Simple badge
    <Badge>Default</Badge>
    
    // With variant
    <Badge variant="success">Success</Badge>
    <Badge variant="warning">Warning</Badge>
    <Badge variant="error">Error</Badge>
    
    // With size
    <Badge size="sm">Small</Badge>
    <Badge size="lg">Large</Badge>
  `,
  
  // Advanced usage examples
  advancedUsage: `
    import { Badge, Icon } from '@acutrack-bookprint/acutrack-ds';
    
    // With icon
    <Badge variant="success" hasIcon icon={<Icon size="xs">check_circle</Icon>}>
      Success
    </Badge>
    
    // Interactive badge
    <Badge variant="primary" isInteractive onClick={() => handleClick()}>
      Clickable Badge
    </Badge>
    
    // Dismissible badge
    <Badge variant="info" isDismissible onDismiss={() => setIsVisible(false)}>
      Dismissible Badge
    </Badge>
    
    // Loading state
    <Badge variant="success" isLoading>
      Loading...
    </Badge>
    
    // With tooltip
    <Badge variant="info" hasTooltip tooltipText="Additional information">
      Hover for Info
    </Badge>
  `,
  
  // Available variants
  variants: {
    default: 'Neutral information',
    primary: 'Main actions, brand elements',
    secondary: 'Secondary information', 
    info: 'Informational content',
    success: 'Positive outcomes, completed states',
    warning: 'Caution, attention needed',
    error: 'Errors, failures, critical issues',
    outline: 'Subtle emphasis',
    ghost: 'Minimal emphasis',
    filled: 'Strong emphasis'
  },
  
  // Available sizes
  sizes: {
    xs: 'Very small labels, inline text (12px)',
    sm: 'Small labels, compact components (14px)',
    md: 'Default size, most common use (16px)',
    lg: 'Medium labels, buttons (18px)',
    xl: 'Large labels, headers (20px)',
    '2xl': 'Extra large labels, hero sections (24px)'
  },
  
  // Key props
  keyProps: {
    variant: 'Visual style variant (default, primary, secondary, info, success, warning, error, outline, ghost, filled)',
    size: 'Badge size (xs, sm, md, lg, xl, 2xl)',
    isInteractive: 'Whether badge is clickable (requires onClick handler)',
    isDismissible: 'Whether badge can be dismissed (requires onDismiss handler)',
    hasIcon: 'Whether badge has an icon (requires icon prop)',
    icon: 'Icon to display (use Icon component)',
    isLoading: 'Whether badge shows loading state (shows spinning refresh icon)',
    isPulsing: 'Whether badge has pulse animation',
    hasTooltip: 'Whether badge has a tooltip (requires tooltipText)',
    tooltipText: 'Tooltip text content'
  },
  
  // Best practices
  bestPractices: {
    do: [
      'Use appropriate variants for different contexts',
      'Provide clear, concise badge text',
      'Use icons to enhance meaning when helpful',
      'Make interactive badges clearly clickable',
      'Use dismissible badges for temporary notifications',
      'Provide tooltips for additional context when needed'
    ],
    dont: [
      'Use too many badges in a single view',
      'Make badge text too long',
      'Use badges for primary actions (use buttons instead)',
      'Mix different badge styles inconsistently',
      'Use badges without considering accessibility'
    ]
  },
  
  // Common use cases
  useCases: {
    statusIndicators: '<Badge variant="success">Active</Badge>',
    categories: '<Badge variant="secondary">Frontend</Badge>',
    notifications: '<Badge variant="error" isDismissible>New Message</Badge>',
    counters: '<Badge variant="primary" size="sm">{count}</Badge>',
    featureFlags: '<Badge variant="success" hasIcon icon={<Icon size="xs">star</Icon>}>Premium</Badge>'
  },
  
  // Design system integration
  designSystem: {
    colors: 'Uses design system colors automatically (system-success, system-warning, system-error, brand-orange, etc.)',
    darkMode: 'Automatically supports dark mode with proper color variants',
    accessibility: 'Includes proper ARIA attributes, keyboard navigation, and screen reader support',
    consistency: 'Follows design system spacing, typography, and interaction patterns'
  }
};
