// @acutrack-bookprint/acutrack-ds/dist/docs/api/ButtonDocumentation.ts
export default {
  componentName: 'Button',
  description: 'A comprehensive button component with 16 variants, 5 sizes, and extensive customization options for building interactive interfaces.',
  
  purpose: 'The Button component provides a consistent way to create interactive elements with various visual styles, states, and behaviors. It supports icons, loading states, button groups, and can be customized for different background contexts.',
  
  props: {
    children: 'React.ReactNode - The content to display inside the button',
    className: 'string - Additional CSS classes to apply to the button',
    variant: "'primary' | 'secondary' | 'text' | 'ghost' | 'outline' | 'dashed' | 'warning' | 'danger' | 'info' | 'success' | 'grey-primary' | 'grey-secondary' | 'grey-text' | 'grey-ghost' | 'grey-outline' | 'grey-dashed' - The visual style variant of the button",
    size: "'xs' | 'sm' | 'md' | 'lg' | 'xl' - The size of the button",
    leftIcon: 'React.ReactNode - Icon to display on the left side of the button content',
    rightIcon: 'React.ReactNode - Icon to display on the right side of the button content',
    onDarkBackground: 'boolean - Whether the button is displayed on a dark background',
    onBrandBackground: 'boolean - Whether the button is displayed on a brand-colored background',
    isDisabled: 'boolean - Whether the button is disabled',
    isLoading: 'boolean - Whether the button is in a loading state',
    isFullWidth: 'boolean - Whether the button should take full width of its container',
    width: "'sm' | 'md' | 'lg' | 'full' | 'auto' | 'none' - Width of the button",
    minWidth: 'string - Minimum width override',
    maxWidth: 'string - Maximum width override',
    hasHoverEffect: 'boolean - Whether the button has hover effects (default: true)',
    hasActiveEffect: 'boolean - Whether the button has active effects (default: false)',
    radius: "'none' | 'sm' | 'md' | 'lg' | 'xl' | 'full' - The border radius of the button",
    as: 'React.ElementType - The HTML element or React component to render as (polymorphic)'
  },
  
  variants: {
    // Core Variants (7)
    primary: 'Orange background with white text - main call-to-action buttons',
    secondary: 'Light background with navy text - secondary actions',
    text: 'Text-only button with no background - subtle actions',
    ghost: 'Transparent background with subtle hover - minimal actions',
    outline: 'Bordered button with transparent background - alternative actions',
    dashed: 'Dashed border button - alternative actions with dashed styling',
    warning: 'Yellow background - warning or caution actions',
    danger: 'Red background - destructive or dangerous actions',
    
    // Extended Variants (9)
    info: 'Blue background - informational actions',
    success: 'Green background - success or positive actions',
    'grey-primary': 'Grey primary variant - neutral primary actions',
    'grey-secondary': 'Grey secondary variant - neutral secondary actions',
    'grey-text': 'Grey text variant - neutral text actions',
    'grey-ghost': 'Grey ghost variant - neutral ghost actions',
    'grey-outline': 'Grey outline variant - neutral outline actions',
    'grey-dashed': 'Grey dashed variant - neutral dashed actions'
  },
  
  sizes: {
    xs: 'Extra small - py-1 px-2 text-xs',
    sm: 'Small - py-1.5 px-3 text-sm',
    md: 'Medium - py-2 px-4 text-base (default)',
    lg: 'Large - py-2.5 px-5 text-lg',
    xl: 'Extra large - py-3 px-6 text-xl'
  },
  
  usage: {
    basic: `<Button onClick={handleClick}>Click Me</Button>`,
    withVariant: `<Button variant="secondary" onClick={handleClick}>Secondary Action</Button>`,
    withSize: `<Button size="lg" onClick={handleClick}>Large Button</Button>`,
    withIcons: `<Button leftIcon={<Icon>star</Icon>} rightIcon={<Icon>arrow_forward</Icon>}>With Icons</Button>`,
    loading: `<Button isLoading={true}>Loading...</Button>`,
    disabled: `<Button isDisabled={true}>Disabled</Button>`,
    fullWidth: `<Button isFullWidth={true}>Full Width</Button>`,
    withEffects: `<Button hasHoverEffect={true} hasActiveEffect={true}>With Effects</Button>`,
    onDarkBackground: `<Button onDarkBackground={true}>On Dark</Button>`,
    polymorphic: `<Button as="a" href="/link">Link Button</Button>`,
    
    // New Variants
    dashed: `<Button variant="dashed" onClick={handleClick}>Dashed Button</Button>`,
    info: `<Button variant="info" onClick={handleClick}>Info Button</Button>`,
    success: `<Button variant="success" onClick={handleClick}>Success Button</Button>`,
    greyPrimary: `<Button variant="grey-primary" onClick={handleClick}>Grey Primary</Button>`,
    greySecondary: `<Button variant="grey-secondary" onClick={handleClick}>Grey Secondary</Button>`,
    greyText: `<Button variant="grey-text" onClick={handleClick}>Grey Text</Button>`,
    greyGhost: `<Button variant="grey-ghost" onClick={handleClick}>Grey Ghost</Button>`,
    greyOutline: `<Button variant="grey-outline" onClick={handleClick}>Grey Outline</Button>`,
    greyDashed: `<Button variant="grey-dashed" onClick={handleClick}>Grey Dashed</Button>`,
    
    // Width Control
    widthControl: `<Button width="lg" onClick={handleClick}>Large Width</Button>`,
    customWidth: `<Button minWidth="200px" maxWidth="300px" onClick={handleClick}>Custom Width</Button>`,
    // customWidth + single-element child (e.g. Avatar): Button does not truncate; overflow-visible so avatar displays fully (v1.0.93)
    customWidthAvatar: `<Button customWidth="40px" variant="ghost"><Avatar size="sm" name="User" /></Button>`,
    
    // Radius Control
    radiusControl: `<Button radius="full" onClick={handleClick}>Pill Button</Button>`,
    
    // Icon Only
    iconOnly: `<Button variant="ghost" size="sm"><Icon>home</Icon></Button>`
  },
  
  examples: {
    primaryActions: `
      <div className="flex gap-3">
        <Button variant="primary">Save Changes</Button>
        <Button variant="secondary">Cancel</Button>
      </div>
    `,
    
    allVariants: `
      <div className="grid grid-cols-2 md:grid-cols-4 lg:grid-cols-8 gap-3">
        <Button variant="primary">Primary</Button>
        <Button variant="secondary">Secondary</Button>
        <Button variant="text">Text</Button>
        <Button variant="ghost">Ghost</Button>
        <Button variant="outline">Outline</Button>
        <Button variant="dashed">Dashed</Button>
        <Button variant="warning">Warning</Button>
        <Button variant="danger">Danger</Button>
        <Button variant="info">Info</Button>
        <Button variant="success">Success</Button>
        <Button variant="grey-primary">Grey Primary</Button>
        <Button variant="grey-secondary">Grey Secondary</Button>
        <Button variant="grey-text">Grey Text</Button>
        <Button variant="grey-ghost">Grey Ghost</Button>
        <Button variant="grey-outline">Grey Outline</Button>
        <Button variant="grey-dashed">Grey Dashed</Button>
      </div>
    `,
    
    iconButtons: `
      <div className="flex gap-2">
        <Button variant="ghost" size="sm">
          <Icon size="sm">edit</Icon>
        </Button>
        <Button variant="ghost" size="sm">
          <Icon size="sm">delete</Icon>
        </Button>
        <Button variant="primary" leftIcon={<Icon>download</Icon>}>
          Download
        </Button>
        <Button variant="outline" rightIcon={<Icon>arrow_forward</Icon>}>
          Continue
        </Button>
      </div>
    `,
    
    loadingStates: `
      <div className="flex gap-3">
        <Button variant="primary" isLoading={true}>Saving...</Button>
        <Button variant="secondary" isDisabled={true}>Processing</Button>
        <Button variant="success" isLoading={true}>Uploading...</Button>
      </div>
    `,
    
    backgroundContext: `
      <div className="p-6 bg-brand-navy rounded-lg">
        <Button variant="primary" onDarkBackground={true}>On Dark</Button>
        <Button variant="outline" onDarkBackground={true}>Outline on Dark</Button>
        <Button variant="text" onDarkBackground={true}>Text on Dark</Button>
      </div>
    `,
    
    formActions: `
      <div className="flex justify-end gap-3">
        <Button variant="text" onClick={handleCancel}>Cancel</Button>
        <Button variant="primary" onClick={handleSubmit}>Submit</Button>
      </div>
    `,
    
    destructiveActions: `
      <div className="flex gap-3">
        <Button variant="danger" onClick={handleDelete}>Delete</Button>
        <Button variant="warning" onClick={handleWarn}>Warning</Button>
        <Button variant="danger" leftIcon={<Icon>delete</Icon>}>
          Delete Account
        </Button>
      </div>
    `,
    
    sizeVariations: `
      <div className="flex flex-wrap gap-3 items-center">
        <Button variant="primary" size="xs">Extra Small</Button>
        <Button variant="primary" size="sm">Small</Button>
        <Button variant="primary" size="md">Medium</Button>
        <Button variant="primary" size="lg">Large</Button>
        <Button variant="primary" size="xl">Extra Large</Button>
      </div>
    `,
    
    radiusVariations: `
      <div className="flex flex-wrap gap-3 items-center">
        <Button variant="primary" radius="none">No Radius</Button>
        <Button variant="primary" radius="sm">Small</Button>
        <Button variant="primary" radius="md">Medium</Button>
        <Button variant="primary" radius="lg">Large</Button>
        <Button variant="primary" radius="xl">Extra Large</Button>
        <Button variant="primary" radius="full">Pill</Button>
      </div>
    `,
    
    widthControl: `
      <div className="space-y-3">
        <Button variant="primary" width="sm">Small Width</Button>
        <Button variant="primary" width="md">Medium Width</Button>
        <Button variant="primary" width="lg">Large Width</Button>
        <Button variant="primary" width="full">Full Width</Button>
        <Button variant="primary" minWidth="200px" maxWidth="300px">
          Custom Width
        </Button>
      </div>
    `,
    
    semanticActions: `
      <div className="space-y-4">
        <div className="flex gap-3">
          <Button variant="success" leftIcon={<Icon>check</Icon>}>
            Success Action
          </Button>
          <Button variant="info" leftIcon={<Icon>info</Icon>}>
            Information
          </Button>
        </div>
        <div className="flex gap-3">
          <Button variant="warning" leftIcon={<Icon>warning</Icon>}>
            Warning Action
          </Button>
          <Button variant="danger" leftIcon={<Icon>error</Icon>}>
            Error Action
          </Button>
        </div>
      </div>
    `,
    
    greyTheme: `
      <div className="grid grid-cols-2 md:grid-cols-3 gap-3">
        <Button variant="grey-primary">Grey Primary</Button>
        <Button variant="grey-secondary">Grey Secondary</Button>
        <Button variant="grey-text">Grey Text</Button>
        <Button variant="grey-ghost">Grey Ghost</Button>
        <Button variant="grey-outline">Grey Outline</Button>
        <Button variant="grey-dashed">Grey Dashed</Button>
      </div>
    `
  },
  
  bestPractices: [
    'Use primary variant for main call-to-action buttons',
    'Use secondary variant for secondary actions',
    'Use text or ghost variants for subtle actions',
    'Use danger variant sparingly for destructive actions',
    'Use info variant for informational actions',
    'Use success variant for positive confirmations',
    'Use warning variant for cautionary actions',
    'Use grey variants for neutral designs or secondary themes',
    'Provide clear, actionable text in button labels',
    'Use icons to enhance button meaning when appropriate',
    'Consider button hierarchy in forms and interfaces',
    'Use loading states for async operations',
    'Disable buttons when actions are not available',
    'Test button accessibility with keyboard navigation',
    'Hover effects are enabled by default for better UX - disable only when needed',
    'Choose appropriate sizes based on context and importance',
    'Use radius="full" for pill-shaped buttons in modern designs',
    'Use width props for consistent button sizing across forms',
    'Use onDarkBackground and onBrandBackground for proper contrast',
    'Group related buttons using ButtonGroup component'
  ],
  
  accessibility: [
    'Buttons are keyboard accessible by default',
    'Focus states are clearly visible with ring indicators',
    'Disabled buttons are properly announced to screen readers',
    'Loading states should be communicated to assistive technology',
    'Button text should be descriptive and actionable',
    'Use aria-label for icon-only buttons',
    'Consider button groups for related actions'
  ],
  
  designSystemIntegration: {
    colors: 'Uses design system color tokens exclusively (brand-orange, brand-navy, system colors)',
    spacing: 'Consistent padding and margins using design system spacing scale',
    typography: 'Uses design system font weights and sizes',
    darkMode: 'Full dark mode support with appropriate color variants',
    focusStates: 'Consistent focus ring styling across all variants',
    transitions: 'Optional smooth transitions for hover and active states'
  },
  
  commonPatterns: {
    formSubmission: 'Primary button for submit, secondary for cancel',
    navigation: 'Text or ghost buttons for navigation links',
    actions: 'Icon buttons for toolbar actions',
    confirmation: 'Danger variant for destructive confirmations',
    loading: 'Loading state during async operations',
    disabled: 'Disabled state when actions are unavailable'
  },
  
  buttonGroup: {
    description: 'ButtonGroup component for grouping related buttons together',
    props: {
      children: 'React.ReactNode - The buttons to group together',
      orientation: "'horizontal' | 'vertical' - The orientation of the button group (default: horizontal)",
      isAttached: 'boolean - Whether the buttons are attached with no gap (default: false)',
      size: "'xs' | 'sm' | 'md' | 'lg' | 'xl' - The size of all buttons in the group (default: md)",
      className: 'string - Additional CSS classes to apply to the button group',
      isDisabled: 'boolean - Whether the group is disabled (disables all buttons)',
      isFullWidth: 'boolean - Whether the group is full width'
    },
    usage: {
      basic: `
        <ButtonGroup>
          <Button variant="primary">First</Button>
          <Button variant="secondary">Second</Button>
          <Button variant="outline">Third</Button>
        </ButtonGroup>
      `,
      attached: `
        <ButtonGroup isAttached>
          <Button variant="primary">First</Button>
          <Button variant="secondary">Second</Button>
          <Button variant="outline">Third</Button>
        </ButtonGroup>
      `,
      vertical: `
        <ButtonGroup orientation="vertical">
          <Button variant="primary">Option 1</Button>
          <Button variant="secondary">Option 2</Button>
          <Button variant="outline">Option 3</Button>
        </ButtonGroup>
      `,
      compound: `
        <ButtonGroup>
          <ButtonGroup.Button variant="primary">Button 1</ButtonGroup.Button>
          <ButtonGroup.Button variant="secondary">Button 2</ButtonGroup.Button>
        </ButtonGroup>
      `
    }
  },
  
  implementationGuide: {
    installation: `
      import { Button, ButtonGroup } from '@acutrack-bookprint/acutrack-ds';
      import { Icon } from '@acutrack-bookprint/acutrack-ds';
    `,
    
    basicImplementation: `
      // Basic button
      <Button onClick={() => console.log('Clicked!')}>
        Click Me
      </Button>
      
      // With variant
      <Button variant="secondary" onClick={handleClick}>
        Secondary Action
      </Button>
      
      // With size
      <Button size="lg" onClick={handleClick}>
        Large Button
      </Button>
    `,
    
    advancedImplementation: `
      // With icons
      <Button 
        variant="primary" 
        leftIcon={<Icon>download</Icon>}
        rightIcon={<Icon>arrow_forward</Icon>}
        onClick={handleDownload}
      >
        Download File
      </Button>
      
      // Loading state
      <Button 
        variant="primary" 
        isLoading={isLoading}
        onClick={handleSubmit}
      >
        {isLoading ? 'Saving...' : 'Save Changes'}
      </Button>
      
      // Custom styling
      <Button 
        variant="primary"
        radius="full"
        width="lg"
        minWidth="200px"
        onClick={handleClick}
      >
        Custom Styled Button
      </Button>
    `,
    
    buttonGroupImplementation: `
      // Basic button group
      <ButtonGroup>
        <Button variant="primary">Save</Button>
        <Button variant="secondary">Cancel</Button>
        <Button variant="outline">Reset</Button>
      </ButtonGroup>
      
      // Attached button group
      <ButtonGroup isAttached>
        <Button variant="primary">First</Button>
        <Button variant="secondary">Second</Button>
        <Button variant="outline">Third</Button>
      </ButtonGroup>
      
      // Vertical button group
      <ButtonGroup orientation="vertical" size="lg">
        <Button variant="primary">Option 1</Button>
        <Button variant="secondary">Option 2</Button>
        <Button variant="outline">Option 3</Button>
      </ButtonGroup>
    `,
    
    formImplementation: `
      // Form actions
      <div className="flex justify-end gap-3">
        <Button variant="text" onClick={handleCancel}>
          Cancel
        </Button>
        <Button variant="primary" onClick={handleSubmit}>
          Submit
        </Button>
      </div>
      
      // Form with loading state
      <div className="flex justify-end gap-3">
        <Button 
          variant="text" 
          isDisabled={isSubmitting}
          onClick={handleCancel}
        >
          Cancel
        </Button>
        <Button 
          variant="primary" 
          isLoading={isSubmitting}
          onClick={handleSubmit}
        >
          {isSubmitting ? 'Submitting...' : 'Submit'}
        </Button>
      </div>
    `,
    
    responsiveImplementation: `
      // Responsive button sizing
      <Button 
        variant="primary"
        size={{ base: 'sm', md: 'md', lg: 'lg' }}
        onClick={handleClick}
      >
        Responsive Button
      </Button>
      
      // Mobile-friendly full width
      <Button 
        variant="primary"
        isFullWidth={{ base: true, md: false }}
        onClick={handleClick}
      >
        Mobile Full Width
      </Button>
    `,
    
    accessibilityImplementation: `
      // Accessible button with proper labels
      <Button 
        variant="danger"
        leftIcon={<Icon>delete</Icon>}
        onClick={handleDelete}
        aria-label="Delete selected item"
      >
        Delete
      </Button>
      
      // Icon-only button with aria-label
      <Button 
        variant="ghost"
        size="sm"
        onClick={handleEdit}
        aria-label="Edit item"
      >
        <Icon>edit</Icon>
      </Button>
    `
  },
  
  migration: {
    fromV1: 'hasHoverEffect now defaults to true for better UX, hasActiveEffect remains false',
    breakingChanges: 'None - all existing props remain compatible',
    newFeatures: [
      '16 button variants including grey theme variants',
      'Extended semantic variants (info, success)',
      'Width control with width, minWidth, maxWidth props',
      'Radius control with radius prop',
      'ButtonGroup component for grouping buttons',
      'Enhanced icon support with proper sizing',
      'Improved dark mode and background context support'
    ]
  }
};
