// @acutrack-bookprint/acutrack-ds/dist/docs/api/DropdownComponentsDocumentation.ts
export default {
  title: 'Dropdown Components Documentation',
  description: 'Comprehensive guide for Select, Autocomplete, and DatePicker components',
  version: '1.0.92',
  lastUpdated: '2026-02-25',
  
  components: {
    Select: {
      purpose: 'A dropdown select component for choosing from a list of predefined options',
      usage: 'Use when you need users to select from a limited set of options',
      
      props: {
        label: { type: 'string', required: true, description: 'Label text displayed above the select field' },
        id: { type: 'string', required: true, description: 'Unique identifier for the select field' },
        options: { type: 'SelectOption[]', required: true, description: 'Array of options to display' },
        value: { type: 'string | number', required: true, description: 'Currently selected value' },
        onChange: { type: 'function', required: true, description: 'Callback when selection changes' },
        size: { type: 'SelectSize', default: 'md', description: 'Size of the select field' },
        variant: { type: 'SelectVariant', default: 'primary', description: 'Visual variant' },
        color: { type: 'SelectColor', default: 'primary', description: 'Color theme' },
        placeholder: { type: 'string', default: 'Select an option...', description: 'Placeholder text' },
        isDisabled: { type: 'boolean', default: false, description: 'Whether the select is disabled' },
        isClearable: { type: 'boolean', default: false, description: 'Whether to show clear button' },
        showDescriptions: { type: 'boolean', default: false, description: 'Whether to show option descriptions' },
        maxDropdownHeight: { type: 'string', default: '15rem', description: 'Maximum height of dropdown' },
        error: { type: 'string', description: 'Error message to display' }
      },
      
      examples: {
        basic: {
          title: 'Basic Select',
          code: `import { Select } from '@acutrack-bookprint/acutrack-ds';

const options = [
  { value: 'option1', label: 'Option 1' },
  { value: 'option2', label: 'Option 2' },
  { value: 'option3', label: 'Option 3' }
];

<Select
  label="Choose an option"
  id="basic-select"
  options={options}
  value={selectedValue}
  onChange={setSelectedValue}
/>`
        },
        
        withDescriptions: {
          title: 'Select with Descriptions',
          code: `const options = [
  { value: 'apple', label: 'Apple', description: 'A red fruit' },
  { value: 'banana', label: 'Banana', description: 'A yellow fruit' }
];

<Select
  label="Choose a fruit"
  id="fruit-select"
  options={options}
  value={selectedValue}
  onChange={setSelectedValue}
  showDescriptions={true}
/>`
        },
        
        clearable: {
          title: 'Clearable Select',
          code: `<Select
  label="Clearable Select"
  id="clearable-select"
  options={options}
  value={selectedValue}
  onChange={setSelectedValue}
  isClearable={true}
/>`
        }
      },
      
      bestPractices: [
        'Use clear, descriptive labels',
        'Provide helpful placeholder text',
        'Use option descriptions for complex choices',
        'Consider using clearable selects for optional fields',
        'Group related options when you have many choices',
        'Use appropriate sizes for your layout context'
      ],
      
      accessibility: [
        'Properly labeled with associated label element',
        'Full keyboard list navigation when list is open: Arrow Up/Down move highlighted option, Enter selects, Home/End jump to first/last option',
        'Focus remains on the trigger (input or button) so typing works in typeable (allowTyping) mode; listbox uses aria-activedescendant and tabIndex=-1 so Tab does not crawl options',
        'Tab / Shift+Tab close the list and move focus to next/previous focusable; Escape closes list and returns focus to trigger',
        'Keyboard-highlighted option uses same visual style as hover (Dropdown.Item isHighlighted); options with semantic colors (success, warning, error, info) show matching background and text when keyboard-highlighted',
        'Arrow and list-navigation keys do not scroll the page (preventDefault at Dropdown Trigger and Content)',
        'Screen reader announcements for selection changes; ARIA attributes for listbox state'
      ]
    },
    
    Autocomplete: {
      purpose: 'An input field with type-ahead suggestions and filtering capabilities',
      usage: 'Use when users need to search through a large list of options or create new ones',
      
      props: {
        label: { type: 'string', required: true, description: 'Label text displayed above the input' },
        id: { type: 'string', required: true, description: 'Unique identifier for the input' },
        options: { type: 'SelectOption[]', required: true, description: 'Array of options to search through' },
        value: { type: 'string | number', required: true, description: 'Currently selected value' },
        onChange: { type: 'function', required: true, description: 'Callback when selection changes' },
        placeholder: { type: 'string', default: 'Type to search...', description: 'Placeholder text' },
        hasFiltering: { type: 'boolean', default: true, description: 'Whether to filter options as you type' },
        showCreateOption: { type: 'boolean', default: false, description: 'Whether to allow creating new options' },
        showDescriptions: { type: 'boolean', default: false, description: 'Whether to show option descriptions' },
        showIcons: { type: 'boolean', default: false, description: 'Whether to show option icons' },
        maxOptions: { type: 'number', default: 10, description: 'Maximum number of options to display' },
        minCharsToFilter: { type: 'number', default: 1, description: 'Minimum characters to start filtering' }
      },
      
      examples: {
        basic: {
          title: 'Basic Autocomplete',
          code: `import { Autocomplete } from '@acutrack-bookprint/acutrack-ds';

const options = [
  { value: 'apple', label: 'Apple' },
  { value: 'banana', label: 'Banana' },
  { value: 'cherry', label: 'Cherry' }
];

<Autocomplete
  label="Search fruits"
  id="fruit-search"
  options={options}
  value={selectedValue}
  onChange={setSelectedValue}
  placeholder="Type to search fruits..."
/>`
        },
        
        withCreate: {
          title: 'Autocomplete with Create Option',
          code: `<Autocomplete
  label="Search or create"
  id="create-search"
  options={options}
  value={selectedValue}
  onChange={setSelectedValue}
  showCreateOption={true}
  createOptionText="Create '{query}'"
  onCreateOption={(value) => console.log('Creating:', value)}
/>`
        },
        
        withIcons: {
          title: 'Autocomplete with Icons',
          code: `const optionsWithIcons = options.map(option => ({
  ...option,
  icon: <Icon size="sm">favorite</Icon>
}));

<Autocomplete
  label="Search with icons"
  id="icon-search"
  options={optionsWithIcons}
  value={selectedValue}
  onChange={setSelectedValue}
  showIcons={true}
/>`
        }
      },
      
      bestPractices: [
        'Provide clear placeholder text indicating search functionality',
        'Use appropriate minimum character thresholds for filtering',
        'Limit the number of displayed options to avoid overwhelming users',
        'Consider using create options for dynamic data entry',
        'Provide visual feedback during search operations',
        'Use icons to enhance option recognition when helpful'
      ],
      
      accessibility: [
        'Proper labeling and association with input field',
        'Keyboard navigation through filtered results',
        'Screen reader announcements for search results',
        'Clear indication of current selection',
        'Escape key to close dropdown and clear search'
      ]
    },
    
    DatePicker: {
      purpose: 'A date selection component with calendar popover and date constraints',
      usage: 'Use when users need to select specific dates with optional constraints',
      
      props: {
        label: { type: 'string', description: 'Label text for the date picker' },
        id: { type: 'string', description: 'Unique identifier for the date picker' },
        value: { type: 'Date | null', required: true, description: 'Currently selected date' },
        onChange: { type: 'function', required: true, description: 'Callback when date changes' },
        size: { type: 'DatePickerSize', default: 'md', description: 'Size of the date picker' },
        variant: { type: 'DatePickerVariant', default: 'outline', description: 'Visual variant' },
        color: { type: 'DatePickerColor', default: 'primary', description: 'Color theme' },
        placeholder: { type: 'string', default: 'Select date', description: 'Placeholder text' },
        hasClearButton: { type: 'boolean', default: true, description: 'Whether to show clear button' },
        hasCalendarIcon: { type: 'boolean', default: true, description: 'Whether to show calendar icon' },
        hasTodayButton: { type: 'boolean', default: true, description: 'Whether to show today button in popover' },
        hasPopover: { type: 'boolean', default: true, description: 'Whether to use popover or inline calendar' },
        minDate: { type: 'Date', description: 'Minimum selectable date' },
        maxDate: { type: 'Date', description: 'Maximum selectable date' },
        closeOnSelect: { type: 'boolean', default: true, description: 'Whether to close popover after selection' }
      },
      
      examples: {
        basic: {
          title: 'Basic DatePicker',
          code: `import { DatePicker } from '@acutrack-bookprint/acutrack-ds';

<DatePicker
  label="Select a date"
  id="basic-datepicker"
  value={selectedDate}
  onChange={setSelectedDate}
  placeholder="Choose a date..."
/>`
        },
        
        withConstraints: {
          title: 'DatePicker with Constraints',
          code: `const today = new Date();
const minDate = new Date(today.getTime() - 30 * 24 * 60 * 60 * 1000); // 30 days ago
const maxDate = new Date(today.getTime() + 30 * 24 * 60 * 60 * 1000); // 30 days from now

<DatePicker
  label="Select date (30 days range)"
  id="constrained-datepicker"
  value={selectedDate}
  onChange={setSelectedDate}
  minDate={minDate}
  maxDate={maxDate}
/>`
        },
        
        inline: {
          title: 'Inline DatePicker',
          code: `<DatePicker
  label="Inline Calendar"
  id="inline-datepicker"
  value={selectedDate}
  onChange={setSelectedDate}
  hasPopover={false}
/>`
        }
      },
      
      bestPractices: [
        'Use appropriate date constraints to guide user selection',
        'Provide clear labels indicating the purpose of date selection',
        'Consider using today button for quick access to current date',
        'Use clear button for optional date fields',
        'Choose appropriate size based on your layout context',
        'Consider using inline calendar for date range selections'
      ],
      
      accessibility: [
        'Proper labeling and association with input field',
        'Keyboard navigation through calendar grid',
        'Screen reader announcements for date selection',
        'Clear indication of current date and selected date',
        'Escape key to close calendar popover'
      ]
    }
  },
  
  commonIssues: {
    dropdownOverflow: {
      title: 'Dropdown Overflow Issues',
      description: 'Dropdowns being clipped by parent containers',
      solution: 'All dropdown components now use z-index: 9999 to ensure they appear above all containers. This has been fixed in version 1.0.54.',
      code: `// Fixed automatically - no code changes needed
// Dropdowns now use z-[9999] class`
    },
    
    hardcodedColors: {
      title: 'Hardcoded Colors',
      description: 'Components using hardcoded colors instead of design system tokens',
      solution: 'All hardcoded colors have been replaced with design system color tokens. Components now properly support dark mode and theme consistency.',
      code: `// Before (hardcoded)
className="bg-blue-500 text-white"

// After (design system)
className="bg-system-info text-system-info-light-text dark:text-system-info-dark-text"`
    },
    
    containerClipping: {
      title: 'Container Clipping',
      description: 'Dropdowns being clipped by parent containers with overflow: hidden',
      solution: 'Use the container overflow test in the test pages to verify dropdown behavior. Consider adjusting parent container overflow settings if needed.',
      code: `// If you need to prevent clipping in your containers
.container {
  overflow: visible; /* Instead of hidden */
}`
    }
  },
  
  testing: {
    testPages: [
      {
        name: 'Select Component Test',
        url: 'select-test.html',
        description: 'Test Select component with various configurations and overflow behavior'
      },
      {
        name: 'Autocomplete Component Test', 
        url: 'autocomplete-test.html',
        description: 'Test Autocomplete component with filtering and suggestions'
      },
      {
        name: 'DatePicker Component Test',
        url: 'datepicker-test.html', 
        description: 'Test DatePicker component with calendar popover and constraints'
      }
    ],
    
    testScenarios: [
      'Test dropdown overflow in card containers',
      'Test all size variants (xs, sm, md, lg, xl)',
      'Test all color variants (primary, success, warning, danger, info)',
      'Test disabled and error states',
      'Test keyboard navigation',
      'Test dark mode compatibility',
      'Test responsive behavior on different screen sizes'
    ]
  },
  
  migration: {
    fromVersion: '1.0.53',
    toVersion: '1.0.54',
    changes: [
      'Fixed dropdown overflow issues with z-index: 9999',
      'Replaced all hardcoded colors with design system tokens',
      'Improved dark mode support across all dropdown components',
      'Added comprehensive test pages for all dropdown components',
      'Enhanced accessibility features and keyboard navigation'
    ],
    
    breakingChanges: [],
    
    deprecations: [],
    
    newFeatures: [
      'Container overflow test scenarios in test pages',
      'Enhanced error state styling with design system colors',
      'Improved focus states and accessibility',
      'Better keyboard navigation support'
    ]
  }
};
