// @acutrack-bookprint/acutrack-ds/dist/docs/api/FooterDocumentation.ts
// Footer Component Documentation
// This file provides comprehensive documentation for the Footer component

export const FooterDocumentation = {
  title: 'Footer Component',
  description: 'Comprehensive footer component with compound structure, theme support, and industry-standard features',
  version: '1.0.66+',
  
  // Quick Start
  quickStart: `
    import { Footer } from '@acutrack-bookprint/acutrack-ds';
    
    // Basic Footer
    <Footer>
      <Footer.Container>
        <div className="text-center">
          <Footer.Copyright company="My Company" />
        </div>
      </Footer.Container>
    </Footer>
    
    // Extended Footer with Theme
    <Footer theme="brand" variant="extended">
      <Footer.Container>
        <div className="grid grid-cols-1 md:grid-cols-4 gap-8">
          <Footer.Brand text="Acutrack" />
          <Footer.Section title="Quick Links">
            <Footer.Links links={links} />
          </Footer.Section>
          <Footer.Section title="Contact">
            <Footer.Contact email="hello@acutrack.com" />
          </Footer.Section>
          <Footer.Newsletter />
        </div>
        <div className="mt-8 pt-8 border-t">
          <Footer.Copyright />
          <Footer.Social socialLinks={socialLinks} />
        </div>
      </Footer.Container>
    </Footer>
  `,
  
  // Features
  features: [
    '5 Theme Variants: default, brand, minimal, grey, dark, custom',
    'Smart Logo Integration: Theme-aware logo switching',
    'Compound Structure: Modular sub-components for flexibility',
    'Industry Standards: Newsletter, social media, contact info, copyright',
    'Responsive Design: Mobile-optimized layouts',
    'Accessibility: WCAG 2.1 compliant',
    'Performance: Event-based theme updates and memoized calculations'
  ],
  
  // Theme System
  themes: {
    default: {
      description: 'Design system colors with automatic dark/light mode support',
      usage: '<Footer theme="default">',
      logoBehavior: 'Changes with light/dark mode toggle'
    },
    brand: {
      description: 'Navy background with orange accents',
      usage: '<Footer theme="brand">',
      logoBehavior: 'Always uses dark logos'
    },
    minimal: {
      description: 'Clean white background with grey logos',
      usage: '<Footer theme="minimal">',
      logoBehavior: 'Always uses grey light logos'
    },
    grey: {
      description: 'Dark grey background with grey dark logos',
      usage: '<Footer theme="grey">',
      logoBehavior: 'Always uses grey dark logos'
    },
    dark: {
      description: 'System dark colors',
      usage: '<Footer theme="dark">',
      logoBehavior: 'Always uses dark logos'
    },
    custom: {
      description: 'Fully customizable colors',
      usage: '<Footer theme="custom" customColors={colors}>',
      logoBehavior: 'Uses custom logo colors'
    }
  },
  
  // Sub-Components
  subComponents: {
    'Footer.Container': {
      description: 'Responsive container wrapper with proper padding and max-width',
      usage: '<Footer.Container>Content</Footer.Container>'
    },
    'Footer.Section': {
      description: 'Content sections with optional titles',
      usage: '<Footer.Section title="Quick Links">Content</Footer.Section>'
    },
    'Footer.Brand': {
      description: 'Logo and company branding with smart theme-aware logo switching',
      usage: '<Footer.Brand text="My Company" />'
    },
    'Footer.Links': {
      description: 'Navigation links with grid layout support',
      usage: '<Footer.Links links={links} columns={2} />'
    },
    'Footer.Social': {
      description: 'Social media links with icons',
      usage: '<Footer.Social socialLinks={socialLinks} />'
    },
    'Footer.Contact': {
      description: 'Contact information display',
      usage: '<Footer.Contact email="hello@company.com" phone="+1-555-123-4567" />'
    },
    'Footer.Newsletter': {
      description: 'Email subscription form',
      usage: '<Footer.Newsletter title="Stay Updated" onSubmit={handleSubscribe} />'
    },
    'Footer.Copyright': {
      description: 'Copyright text with year and company',
      usage: '<Footer.Copyright company="My Company" year={2025} />'
    }
  },
  
  // Props
  props: {
    Footer: {
      variant: 'Footer variant (default, minimal, extended, landing, dashboard)',
      size: 'Footer size (sm, md, lg, xl)',
      theme: 'Theme variant (default, brand, minimal, grey, dark, custom)',
      customColors: 'Custom color configuration object',
      sticky: 'Whether footer is sticky to bottom',
      className: 'Additional CSS classes'
    },
    'Footer.Brand': {
      text: 'Company name to display with logo',
      href: 'Link URL for clickable logo',
      isClickable: 'Whether logo is clickable',
      className: 'Additional CSS classes'
    },
    'Footer.Links': {
      links: 'Array of link objects with label, href, and optional external flag',
      columns: 'Number of columns for grid layout',
      className: 'Additional CSS classes'
    },
    'Footer.Social': {
      socialLinks: 'Array of social media objects with name, href, and icon',
      className: 'Additional CSS classes'
    },
    'Footer.Contact': {
      email: 'Email address',
      phone: 'Phone number',
      address: 'Physical address',
      className: 'Additional CSS classes'
    },
    'Footer.Newsletter': {
      title: 'Newsletter section title',
      placeholder: 'Email input placeholder text',
      buttonText: 'Subscribe button text',
      onSubmit: 'Callback function for form submission',
      className: 'Additional CSS classes'
    },
    'Footer.Copyright': {
      text: 'Custom copyright text',
      year: 'Copyright year',
      company: 'Company name',
      className: 'Additional CSS classes'
    }
  },
  
  // Examples
  examples: {
    ecommerce: `
      <Footer theme="brand" variant="extended">
        <Footer.Container>
          <div className="grid grid-cols-1 md:grid-cols-4 gap-8">
            <Footer.Section>
              <Footer.Brand text="ShopName" />
              <p className="mt-4 text-sm">Your one-stop shop for quality products.</p>
              <Footer.Social socialLinks={socialLinks} />
            </Footer.Section>
            
            <Footer.Section title="Shop">
              <Footer.Links links={[
                { label: 'All Products', href: '/products' },
                { label: 'New Arrivals', href: '/new' },
                { label: 'Best Sellers', href: '/bestsellers' },
                { label: 'Sale', href: '/sale' }
              ]} />
            </Footer.Section>
            
            <Footer.Section title="Customer Service">
              <Footer.Links links={[
                { label: 'Help Center', href: '/help' },
                { label: 'Shipping Info', href: '/shipping' },
                { label: 'Returns', href: '/returns' },
                { label: 'Size Guide', href: '/sizing' }
              ]} />
            </Footer.Section>
            
            <Footer.Section title="Stay Connected">
              <Footer.Newsletter 
                title="Newsletter"
                placeholder="Enter your email"
                buttonText="Subscribe"
                onSubmit={handleNewsletter}
              />
              <Footer.Contact 
                email="support@shopname.com"
                phone="1-800-SHOP-NOW"
              />
            </Footer.Section>
          </div>
          
          <div className="mt-12 pt-8 border-t">
            <div className="flex flex-col md:flex-row justify-between items-center">
              <Footer.Copyright company="ShopName" year={2025} />
              <div className="flex space-x-6 mt-4 md:mt-0">
                <a href="/privacy" className="text-sm hover:text-brand-orange">Privacy</a>
                <a href="/terms" className="text-sm hover:text-brand-orange">Terms</a>
                <a href="/cookies" className="text-sm hover:text-brand-orange">Cookies</a>
              </div>
            </div>
          </div>
        </Footer.Container>
      </Footer>
    `,
    
    corporate: `
      <Footer theme="minimal" variant="extended">
        <Footer.Container>
          <div className="grid grid-cols-1 md:grid-cols-3 gap-8">
            <Footer.Section>
              <Footer.Brand text="CompanyName" />
              <p className="mt-4 text-sm">
                Leading the industry with innovative solutions and exceptional service.
              </p>
            </Footer.Section>
            
            <Footer.Section title="Company">
              <Footer.Links links={[
                { label: 'About Us', href: '/about' },
                { label: 'Our Team', href: '/team' },
                { label: 'Careers', href: '/careers' },
                { label: 'Press', href: '/press' }
              ]} />
            </Footer.Section>
            
            <Footer.Section title="Contact">
              <Footer.Contact
                email="info@company.com"
                phone="+1 (555) 123-4567"
                address="123 Business Ave, Suite 100, City, State 12345"
              />
            </Footer.Section>
          </div>
          
          <div className="mt-8 pt-8 border-t text-center">
            <Footer.Copyright company="CompanyName" year={2025} />
          </div>
        </Footer.Container>
      </Footer>
    `,
    
    blog: `
      <Footer theme="default" variant="minimal">
        <Footer.Container>
          <div className="flex flex-col md:flex-row justify-between items-center">
            <Footer.Brand text="BlogName" />
            <div className="flex space-x-6 mt-4 md:mt-0">
              <Footer.Social socialLinks={socialLinks} />
            </div>
          </div>
          <div className="mt-4 text-center">
            <Footer.Copyright company="BlogName" year={2025} />
          </div>
        </Footer.Container>
      </Footer>
    `
  },
  
  // Best Practices
  bestPractices: [
    'Group related links into logical sections',
    'Use clear, descriptive section titles and link labels',
    'Limit each section to 5-7 links maximum',
    'Place most important content first',
    'Use appropriate theme for your brand',
    'Test in both light and dark modes',
    'Ensure proper mobile responsiveness',
    'Use semantic HTML structure',
    'Provide proper ARIA labels for accessibility',
    'Optimize for performance with stable references'
  ],
  
  // Accessibility
  accessibility: [
    'Uses proper <footer> semantic element',
    'Semantic heading structure for screen readers',
    'Proper link attributes and descriptions',
    'Keyboard navigation support',
    'High contrast color ratios (WCAG 2.1 AA)',
    'Focus indicators for interactive elements',
    'Screen reader friendly content structure'
  ],
  
  // Migration Guide
  migration: {
    fromBasicFooter: `
      // Old (Layout.tsx Footer)
      <Footer hasBackground hasElevation>
        <Container size="screen" padding="lg">
          <div className="text-center">
            <p>© 2024 My Company. All rights reserved.</p>
          </div>
        </Container>
      </Footer>
      
      // New (Compound Footer)
      <Footer theme="default" variant="extended">
        <Footer.Container>
          <div className="text-center">
            <Footer.Copyright company="My Company" year={2024} />
          </div>
        </Footer.Container>
      </Footer>
    `,
    
    addingTheme: `
      // Add theme to existing footer
      <Footer theme="brand" variant="extended">
        {/* Existing content */}
      </Footer>
    `
  },
  
  // Troubleshooting
  troubleshooting: {
    'Logo not changing with theme': 'Ensure you\'re using Footer.Brand correctly with text prop',
    'Links not styling correctly': 'Check that links array has correct structure with label and href',
    'Theme colors not applying': 'Verify custom colors are properly formatted in RGB format',
    'Responsive issues': 'Use proper responsive classes like grid-cols-1 md:grid-cols-2 lg:grid-cols-4',
    'Performance issues': 'Use stable references for links and social data, avoid inline object creation'
  }
};

export default FooterDocumentation;
