// @acutrack-bookprint/acutrack-ds/dist/docs/api/HeaderDocumentation.ts
export default {
  title: 'Header Component',
  description: 'Application header with built-in Acutrack logos, navigation, user controls, and responsive design',
  
  // Latest updates (v1.0.68)
  latestUpdates: {
    version: '1.0.68',
    changes: [
      'Fixed user menu icons displaying same icon for all menu items',
      'Replaced hardcoded SVG icons with dynamic Icon component integration',
      'Fixed user menu hover effects using proper design system theme classes',
      'Removed hardcoded values and replaced with theme-consistent styling',
      'Enhanced danger color hover visibility in light mode',
      'Improved user menu interaction with proper visual feedback'
    ]
  },
  
  // Basic usage examples
  basicUsage: `
    import { Header } from '@acutrack-bookprint/acutrack-ds';
    
    // Basic header with default logo
    <Header />
    
    // Header with search
    <Header 
      hasSearch={true}
      searchPlaceholder="Search..."
      onSearch={(query) => console.log('Search:', query)}
    />
    
    // Header with navigation
    <Header 
      links={[
        { label: 'Home', href: '/' },
        { label: 'About', href: '/about' },
        { label: 'Contact', href: '/contact' }
      ]}
    />
  `,
  
  // Advanced usage examples
  advancedUsage: `
    import { Header, DEFAULT_LOGOS, Icon } from '@acutrack-bookprint/acutrack-ds';
    
    // Header with custom logo
    <Header logo={DEFAULT_LOGOS.light} />
    
    // Header with user controls (Updated v1.0.68)
    <Header 
      user={{
        name: 'John Doe',
        email: 'john@example.com',
        avatar: '/avatar.jpg',
        role: 'Administrator'
      }}
      userMenuLinks={[
        { href: '#profile', label: 'Profile', icon: 'person' },
        { href: '#settings', label: 'Settings', icon: 'settings' },
        { href: '#help', label: 'Help & Support', icon: 'help' },
        { href: '#logout', label: 'Logout', icon: 'logout', color: 'danger' }
      ]}
      onLogout={() => handleLogout()}
    />
    
    // Header with notifications
    <Header 
      notifications={[
        { id: 1, title: 'New message', content: 'You have a new message' },
        { id: 2, title: 'System update', content: 'System will be updated tonight' }
      ]}
      unreadCount={2}
    />
    
    // Header with icons in navigation
    <Header 
      links={[
        { 
          label: 'Dashboard', 
          href: '/dashboard',
          icon: <Icon size="sm">dashboard</Icon>
        },
        { 
          label: 'Settings', 
          href: '/settings',
          icon: <Icon size="sm">settings</Icon>
        }
      ]}
    />
    
    // Hide logo
    <Header logo={{ show: false }} />
  `,
  
  // Logo system
  logoSystem: {
    defaultLogos: 'Built-in Acutrack logos are automatically included',
    logoAssets: `
      import { 
        ACUTRACK_LOGO_LIGHT,
        ACUTRACK_LOGO_DARK,
        ACUTRACK_LOGO_ICON_LIGHT,
        ACUTRACK_LOGO_ICON_DARK,
        DEFAULT_LOGOS,
        DEFAULT_LOGO_CONFIGS
      } from '@acutrack-bookprint/acutrack-ds';
    `,
    logoVisibility: [
      'Show logo: <Header /> (default)',
      'Hide logo: <Header logo={{ show: false }} />',
      'Custom logo: <Header logo={customLogoConfig} />',
      'Conditional: <Header logo={user.isAdmin ? DEFAULT_LOGOS.light : { show: false }} />'
    ]
  },
  
  // Available variants
  variants: {
    website: 'Standard website header with full navigation',
    dashboard: 'Dashboard header with user controls and notifications',
    minimal: 'Minimal header with just logo and essential controls',
    full: 'Full-featured header with all available options'
  },
  
  // Available sizes
  sizes: {
    sm: 'Small header for compact layouts',
    md: 'Default size for most applications',
    lg: 'Large header for prominent branding',
    xl: 'Extra large header for hero sections'
  },
  
  // Key props
  keyProps: {
    logo: 'Logo configuration (use DEFAULT_LOGOS or custom config)',
    links: 'Navigation links array with label, href, and optional icon',
    user: 'User information object (name, email, avatar, role)',
    userMenuLinks: 'User menu links array with href, label, icon, and optional color (v1.0.68+)',
    notifications: 'Notifications array for user alerts',
    unreadCount: 'Number of unread notifications',
    onLogout: 'Logout handler function',
    hasSearch: 'Whether to show search functionality',
    searchPlaceholder: 'Placeholder text for search input',
    onSearch: 'Search handler function',
    variant: 'Header variant (website, dashboard, minimal, full)',
    size: 'Header size (sm, md, lg, xl)',
    className: 'Additional CSS classes'
  },
  
  // Best practices
  bestPractices: {
    do: [
      'Use the default logo configuration for consistency',
      'Provide clear navigation labels and proper href attributes',
      'Include user information for authenticated users',
      'Use appropriate icons in navigation links and user menu items (v1.0.68+)',
      'Handle logout functionality properly',
      'Provide meaningful notification content',
      'Use responsive design considerations',
      'Use proper color coding for user menu items (danger for logout)',
      'Ensure user menu icons are unique and meaningful'
    ],
    dont: [
      'Use external logo URLs (use built-in logos or import assets)',
      'Create navigation links without proper href attributes',
      'Forget to handle user authentication states',
      'Use too many navigation items (keep it focused)',
      'Ignore mobile responsiveness',
      'Use hardcoded user data (make it dynamic)',
      'Use hardcoded SVG icons in user menu (use Icon component)',
      'Forget to provide proper hover effects for user menu items'
    ]
  },
  
  // UserMenu component (v1.0.68+)
  userMenu: {
    description: 'Enhanced user menu with dynamic icons and improved hover effects',
    props: {
      user: {
        type: 'UserConfig',
        description: 'User information object',
        required: true,
        example: '{ name: "John Doe", avatar: "https://example.com/avatar.jpg" }'
      },
      links: {
        type: 'UserMenuLink[]',
        description: 'Array of menu links with icons and colors',
        required: true,
        example: `[
          { href: "#profile", label: "Profile", icon: "person" },
          { href: "#settings", label: "Settings", icon: "settings" },
          { href: "#logout", label: "Logout", icon: "logout", color: "danger" }
        ]`
      }
    },
    features: [
      'Dynamic icon display using design system Icon component',
      'Proper hover effects with theme-consistent styling',
      'Danger color support for destructive actions',
      'Responsive design with mobile support',
      'Accessibility compliant with proper ARIA attributes'
    ],
    hoverEffects: {
      default: 'hover:bg-bg-tertiary dark:hover:bg-dark-bg-tertiary',
      danger: 'hover:bg-system-error-dark hover:text-white dark:hover:bg-system-error-dark dark:hover:text-white'
    }
  },
  logoVisibility: {
    showDefault: '<Header />',
    hideLogo: '<Header logo={{ show: false }} />',
    customLogo: '<Header logo={customLogoConfig} />',
    conditional: '<Header logo={condition ? DEFAULT_LOGOS.light : { show: false }} />',
    responsive: '<Header logo={{ show: true, useIconOnMobile: true }} />'
  },
  
  // Navigation patterns
  navigationPatterns: {
    simple: `
      <Header 
        links={[
          { label: 'Home', href: '/' },
          { label: 'About', href: '/about' }
        ]}
      />
    `,
    withIcons: `
      <Header 
        links={[
          { 
            label: 'Dashboard', 
            href: '/dashboard',
            icon: <Icon size="sm">dashboard</Icon>
          }
        ]}
      />
    `,
    withUser: `
      <Header 
        user={{
          name: 'John Doe',
          email: 'john@example.com',
          avatar: '/avatar.jpg'
        }}
        onLogout={() => handleLogout()}
      />
    `
  },
  
  // Design system integration
  designSystem: {
    colors: 'Uses design system colors (brand-orange, brand-navy, system colors)',
    darkMode: 'Automatically supports dark mode with proper logo switching',
    accessibility: 'Includes proper ARIA attributes, keyboard navigation, and screen reader support',
    consistency: 'Follows design system spacing, typography, and interaction patterns',
    responsive: 'Built-in responsive design with mobile-first approach'
  },
  
  // Common use cases
  useCases: {
    website: `
      <Header 
        variant="website"
        links={[
          { label: 'Home', href: '/' },
          { label: 'Products', href: '/products' },
          { label: 'About', href: '/about' },
          { label: 'Contact', href: '/contact' }
        ]}
        hasSearch={true}
      />
    `,
    dashboard: `
      <Header 
        variant="dashboard"
        user={{
          name: 'John Doe',
          email: 'john@example.com',
          role: 'Administrator'
        }}
        notifications={notifications}
        onLogout={() => handleLogout()}
      />
    `,
    minimal: `
      <Header 
        variant="minimal"
        hasSearch={true}
        searchPlaceholder="Search..."
      />
    `
  },
  
  // Accessibility features
  accessibility: {
    navigation: 'Proper navigation landmarks and ARIA labels',
    keyboardNavigation: 'Full keyboard support for all interactive elements',
    screenReader: 'Proper ARIA attributes and descriptions',
    focusManagement: 'Clear focus indicators and logical tab order',
    colorContrast: 'Meets WCAG contrast requirements'
  },
  
  // Troubleshooting
  troubleshooting: {
    logoNotShowing: [
      'Ensure you\'re using the default logo configuration',
      'Check that logo assets are properly imported',
      'Verify that the logo show property is not set to false'
    ],
    navigationNotWorking: [
      'Ensure links have proper href attributes',
      'Check that the links array is properly formatted',
      'Verify that the navigation is not disabled'
    ],
    userControlsNotShowing: [
      'Ensure user object is properly provided',
      'Check that user properties are correctly formatted',
      'Verify that the header variant supports user controls'
    ],
    searchNotWorking: [
      'Ensure hasSearch is set to true',
      'Check that onSearch handler is provided',
      'Verify that search functionality is not disabled'
    ]
  },
  
  // Size guidelines
  sizeGuidelines: {
    sm: 'Small header for compact layouts, mobile-first designs',
    md: 'Default size for most web applications',
    lg: 'Large header for prominent branding, desktop focus',
    xl: 'Extra large header for hero sections, landing pages'
  },
  
  // Variant guidelines
  variantGuidelines: {
    website: 'Standard website with navigation and search',
    dashboard: 'Application dashboard with user controls and notifications',
    minimal: 'Simple header with essential features only',
    full: 'Full-featured header with all available options'
  }
};
