// @acutrack-bookprint/acutrack-ds/dist/docs/api/index.ts
// Component Documentation Exports
// This file exports all component documentation for AI assistants and developers

// AI Usage Prompts
export const AI_ICON_USAGE_PROMPT = `
🤖 CRITICAL AI INSTRUCTIONS FOR ICON USAGE:

1. ALWAYS use design system Icon component: import { Icon } from '@acutrack-bookprint/acutrack-ds';
2. NEVER use other icon libraries (react-icons, font-awesome, material-icons)
3. NEVER add external icon fonts to HTML
4. Use valid icon names from the 100+ available icons
5. Icons have automatic fallbacks (emoji/unicode) when font not loaded
6. Icons inherit parent text color automatically
7. Complete offline support - no external dependencies

VALID ICONS: 'home', 'settings', 'search', 'menu', 'close', 'add', 'edit', 'delete', 'save', 'cancel', 'check_circle', 'warning', 'error', 'info', 'help', 'favorite', 'person', 'email', 'phone', 'notifications', 'shopping_cart', 'download', 'upload', 'print', 'share', 'arrow_back', 'arrow_forward', 'more_vert', 'more_horiz', 'dashboard', 'palette', 'code', 'security', 'analytics', 'dark_mode', 'light_mode', 'wifi', 'lock', 'star', 'bookmark', 'flag', 'verified', 'payment', 'trending_up', 'schedule', 'play_arrow', 'pause', 'volume_up', 'camera_alt', 'folder', 'file_download', 'content_copy', 'undo', 'redo' and 50+ more.

FALLBACKS: home→🏠, settings→⚙️, search→🔍, menu→☰, close→✕, add→+, edit→✏️, delete→🗑️, save→💾, cancel→❌, check_circle→✅, warning→⚠️, error→❌, info→ℹ️, help→❓, favorite→❤️, person→👤, email→📧, phone→📞, notifications→🔔, shopping_cart→🛒, download→⬇️, upload→⬆️, share→📤, arrow_back→←, arrow_forward→→, more_vert→⋮, more_horiz→⋯, dashboard→📊, code→💻, security→🔒, analytics→📈, dark_mode→🌙, light_mode→☀️, refresh→🔄, lock→🔒, star→⭐, bookmark→🔖, flag→🚩, verified→✅, payment→💳, trending_up→📈, schedule→📅, play_arrow→▶️, pause→⏸️, volume_up→🔊, camera_alt→📷, folder→📁, file_download→📥, content_copy→📋, undo→↶, redo→↷

INVALID ICONS → ❓ (Help icon)

ALWAYS use <Icon>icon_name</Icon> format. NEVER mix with other icon libraries!
`;

export { default as BadgeDocumentation } from './BadgeDocumentation';
export { default as IconDocumentation } from './IconDocumentation';
export { default as AlertDocumentation } from './AlertDocumentation';
export { default as RadioDocumentation } from './RadioDocumentation';
export { default as HeaderDocumentation } from './HeaderDocumentation';
export { default as LayoutDocumentation } from './LayoutDocumentation';
export { default as VirtualizedTableDocumentation } from './VirtualizedTableDocumentation';
export { default as ToastDocumentation } from './ToastDocumentation';
export { default as DropdownComponentsDocumentation } from './DropdownComponentsDocumentation';
export { default as ButtonDocumentation } from './ButtonDocumentation';
export { default as FooterDocumentation } from './FooterDocumentation';

// Re-export component documentation content
export const COMPONENT_DOCUMENTATION = {
  Badge: {
    title: 'Badge Component',
    description: 'Flexible status indicators and labels with multiple variants and interactive states',
    usage: `
      import { Badge } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <Badge variant="success">Success</Badge>
      
      // With icon
      <Badge variant="info" hasIcon icon={<Icon size="xs">info</Icon>}>
        Info
      </Badge>
      
      // Interactive
      <Badge variant="primary" isInteractive onClick={handleClick}>
        Clickable
      </Badge>
      
      // Dismissible
      <Badge variant="warning" isDismissible onDismiss={handleDismiss}>
        Dismissible
      </Badge>
    `,
    variants: ['default', 'primary', 'secondary', 'info', 'success', 'warning', 'error', 'outline', 'ghost', 'filled'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl', '2xl'],
    props: {
      variant: 'Visual style variant',
      size: 'Badge size',
      isInteractive: 'Whether badge is clickable',
      isDismissible: 'Whether badge can be dismissed',
      hasIcon: 'Whether badge has an icon',
      icon: 'Icon to display',
      isLoading: 'Whether badge shows loading state',
      isPulsing: 'Whether badge has pulse animation',
      hasTooltip: 'Whether badge has a tooltip',
      tooltipText: 'Tooltip text content'
    }
  },
  
  Icon: {
    title: 'Icon Component',
    description: 'Material Symbols icons with size, weight, and fill variations for consistent iconography',
    usage: `
      import { Icon } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <Icon>home</Icon>
      
      // With size
      <Icon size="lg">settings</Icon>
      
      // Filled icon
      <Icon size="md" filled>favorite</Icon>
      
      // With weight
      <Icon size="lg" weight={600}>notifications</Icon>
      
      // Interactive icon
      <button aria-label="Settings">
        <Icon size="lg">settings</Icon>
      </button>
      
      // Decorative icon
      <Icon size="sm" ariaHidden={true}>star</Icon>
    `,
    sizes: ['xs', 'sm', 'md', 'lg', 'xl', '2xl', '3xl'],
    props: {
      children: 'Material Symbol name (required) - e.g., "home", "settings"',
      size: 'Icon size (use size prop, not className)',
      filled: 'Whether icon should be filled (1) or outlined (0)',
      weight: 'Font weight (100-700) for icon thickness',
      opticalSize: 'Material Symbols optical size for different contexts',
      className: 'Additional CSS classes',
      ariaLabel: 'Accessible label for interactive icons',
      ariaHidden: 'Hide from screen readers for decorative icons'
    },
    fontRequirement: 'Requires full range Material Symbols font: https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,0..200',
    commonIcons: {
      navigation: ['home', 'menu', 'arrow_back', 'arrow_forward', 'close'],
      actions: ['add', 'edit', 'delete', 'save', 'cancel', 'refresh'],
      status: ['check_circle', 'error', 'warning', 'info', 'help'],
      communication: ['email', 'phone', 'chat', 'notifications'],
      content: ['favorite', 'star', 'bookmark', 'share'],
      system: ['settings', 'account_circle', 'lock', 'visibility']
    }
  },
  
  Alert: {
    title: 'Alert Component',
    description: 'Contextual feedback messages with multiple variants, sizes, and dismissible states for user notifications',
    usage: `
      import { Alert, Icon } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <Alert variant="success" title="Success!" content="Operation completed." />
      
      // Dismissible
      <Alert variant="warning" title="Warning" isDismissible onDismiss={handleDismiss} />
      
      // With custom icon
      <Alert variant="info" customIcon={<Icon size="md">info</Icon>} />
      
      // Different sizes
      <Alert variant="info" size="xs" title="Small Alert" />
      <Alert variant="info" size="lg" title="Large Alert" content="This is a larger alert." />
    `,
    variants: ['primary', 'secondary', 'info', 'success', 'warning', 'error'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    props: {
      variant: 'Alert variant (primary, secondary, info, success, warning, error)',
      size: 'Alert size (xs, sm, md, lg, xl) - affects icon size and padding',
      title: 'Alert title/heading text',
      content: 'Alert content/description text',
      children: 'Rich content instead of simple text content',
      isDismissible: 'Whether alert can be dismissed (shows close button)',
      onDismiss: 'Callback function when alert is dismissed',
      customIcon: 'Custom icon to override default variant icon'
    }
  },
  
  Radio: {
    title: 'Radio Component',
    description: 'Single-choice form input with multiple variants, sizes, and states for user selection',
    usage: `
      import { Radio } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <Radio id="option1" name="choice" value="option1" label="Option 1" />
      
      // Radio group
      <div>
        <Radio id="small" name="size" value="small" label="Small" />
        <Radio id="medium" name="size" value="medium" label="Medium" />
        <Radio id="large" name="size" value="large" label="Large" />
      </div>
      
      // With description
      <Radio 
        id="premium"
        name="plan"
        value="premium"
        label="Premium Plan"
        description="Includes all features and priority support"
      />
    `,
    variants: ['primary', 'success', 'warning', 'danger', 'info'],
    sizes: ['sm', 'md', 'lg', 'xl'],
    props: {
      id: 'Unique identifier for the radio button (required)',
      name: 'Name attribute for grouping radio buttons (required)',
      value: 'Value of the radio button (required)',
      label: 'Label text for the radio button',
      description: 'Optional description text below the label',
      checked: 'Whether the radio button is selected',
      onChange: 'Callback function when selection changes',
      variant: 'Visual variant (primary, success, warning, danger, info)',
      size: 'Size of the radio button (sm, md, lg, xl)',
      isDisabled: 'Whether the radio button is disabled',
      isLoading: 'Whether the radio button is in loading state',
      isRequired: 'Whether the radio button is required'
    }
  },
  
  Header: {
    title: 'Header Component',
    description: 'Application header with built-in Acutrack logos, navigation, user controls, and responsive design',
    usage: `
      import { Header, DEFAULT_LOGOS, Icon } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage with default logo
      <Header />
      
      // Header with search
      <Header 
        hasSearch={true}
        searchPlaceholder="Search..."
        onSearch={(query) => console.log('Search:', query)}
      />
      
      // Header with navigation
      <Header 
        links={[
          { label: 'Home', href: '/' },
          { label: 'About', href: '/about' }
        ]}
      />
      
      // Hide logo
      <Header logo={{ show: false }} />
    `,
    variants: ['website', 'dashboard', 'minimal', 'full'],
    sizes: ['sm', 'md', 'lg', 'xl'],
    props: {
      logo: 'Logo configuration (use DEFAULT_LOGOS or custom config)',
      links: 'Navigation links array with label, href, and optional icon',
      user: 'User information object (name, email, avatar, role)',
      notifications: 'Notifications array for user alerts',
      unreadCount: 'Number of unread notifications',
      onLogout: 'Logout handler function',
      hasSearch: 'Whether to show search functionality',
      searchPlaceholder: 'Placeholder text for search input',
      onSearch: 'Search handler function',
      variant: 'Header variant (website, dashboard, minimal, full)',
      size: 'Header size (sm, md, lg, xl)'
    }
  },
  
  Button: {
    title: 'Button Component',
    description: 'Interactive buttons with multiple variants, sizes, and icon support',
    usage: `
      import { Button, Icon } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <Button variant="primary">Click me</Button>
      
      // With icons
      <Button leftIcon={<Icon size="sm">add</Icon>}>Add Item</Button>
      <Button rightIcon={<Icon size="sm">arrow_forward</Icon>}>Continue</Button>
      
      // Different variants
      <Button variant="secondary">Secondary</Button>
      <Button variant="outline">Outline</Button>
    `,
    variants: ['primary', 'secondary', 'text', 'ghost', 'outline', 'warning', 'danger'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    props: {
      variant: 'Button style variant',
      size: 'Button size',
      leftIcon: 'Icon on the left side',
      rightIcon: 'Icon on the right side',
      isDisabled: 'Whether button is disabled',
      isLoading: 'Whether button shows loading state',
      isFullWidth: 'Whether button takes full width'
    }
  },

  Layout: {
    title: 'Layout Components',
    description: 'Comprehensive layout system for building responsive, screen-friendly interfaces',
    usage: `
      import { Layout, LayoutHeader, Main, Footer, Grid, Card, Container } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic layout structure
      <Layout hasMinHeight isColumn isFullWidth>
        <LayoutHeader isSticky hasElevation>
          <h1>Page Title</h1>
        </LayoutHeader>
        <Main hasPadding>
          <Container size="lg" isCentered>
            <Grid cols={{ xs: 1, sm: 2, md: 3 }} gap={6}>
              <Card isInteractive>
                <Card.Body>Content</Card.Body>
              </Card>
            </Grid>
          </Container>
        </Main>
        <Footer hasBackground>
          <p>© 2024 Acutrack</p>
        </Footer>
      </Layout>
      
      // Responsive grid
      <Grid cols={{ xs: 1, sm: 2, md: 3, lg: 4 }} gap={6}>
        {items.map(item => (
          <Card key={item.id} isInteractive>
            <Card.Body>{item.content}</Card.Body>
          </Card>
        ))}
      </Grid>
      
      // Container sizes
      <Container size="md" variant="article" hasBackground>
        <h1>Article Content</h1>
      </Container>
      
      // Card with overflow visible for dropdowns
      <Card overflowVisible={true}>
        <DatePicker label="Select date" />
      </Card>
    `,
    components: {
      Layout: 'Main layout wrapper with flexible configuration',
      Container: 'Content width management and responsive containers',
      Grid: 'Responsive grid system for content organization',
      Card: 'Flexible content containers with multiple variants',
      LayoutHeader: 'Page header with sticky and elevation options',
      Main: 'Main content area with responsive width management',
      LayoutSidebar: 'Sidebar with responsive width and positioning',
      Footer: 'Page footer with elevation and background options',
      Content: 'Content wrapper with padding and styling options',
      TwoColumnLayout: 'Two-column responsive layout',
      ThreeColumnLayout: 'Three-column responsive layout'
    },
    props: {
      Layout: {
        hasMinHeight: 'Whether layout has minimum viewport height',
        isColumn: 'Whether layout uses flexbox column direction',
        isFullWidth: 'Whether layout is full width',
        variant: 'Layout variant (dashboard, landing, admin, minimal, elevated)',
        hasBackground: 'Whether layout has background',
        hasBorder: 'Whether layout has border',
        hasShadow: 'Whether layout has shadow'
      },
      Container: {
        size: 'Container size (xs, sm, md, lg, xl, 2xl, full, screen)',
        variant: 'Container variant (default, article, main, aside, section, card, panel, modal)',
        isCentered: 'Whether container is centered horizontally',
        padding: 'Padding size (none, sm, md, lg, xl)',
        hasBackground: 'Whether container has background',
        hasBorder: 'Whether container has border'
      },
      Grid: {
        cols: 'Number of columns or responsive configuration object',
        gap: 'Gap size between grid items (1-32)',
        hasBackground: 'Whether grid has background',
        isBordered: 'Whether grid has borders',
        hasShadow: 'Whether grid has shadow',
        isLoading: 'Whether grid is in loading state'
      },
      Card: {
        size: 'Card size (xs, sm, md, lg, xl, 2xl)',
        variant: 'Card variant (default, minimal, elevated, bordered, filled, outlined, ghost, card, panel)',
        isInteractive: 'Whether card has hover effects',
        isFeatured: 'Whether card is featured',
        isBordered: 'Whether card has border',
        hasShadow: 'Whether card has shadow',
        fullWidth: 'Whether card takes full width',
        overflowVisible: 'Whether card allows overflow (for dropdowns, popovers, etc.)'
      }
    }
  },
  
  VirtualizedTable: {
    title: 'VirtualizedTable Component',
    description: 'A high-performance, feature-rich virtualized table component for handling large datasets with extensive customization options.',
    usage: `
      import { VirtualizedTable } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <VirtualizedTable
        data={users}
        columns={columns}
        height={400}
        rowHeight={50}
        searchable={true}
        selectable={true}
        striped={true}
      />
      
      // Advanced usage with customization
      <VirtualizedTable
        data={users}
        columns={columns}
        height={600}
        rowHeight={60}
        size="lg"
        variant="bordered"
        color="primary"
        searchable={true}
        filterable={true}
        selectable={true}
        multiSort={true}
        striped={true}
        hoverable={true}
        showHeader={true}
        showFooter={true}
        showRowNumbers={true}
        highlightedRows={['user-1', 'user-5']}
        highlightColor="success"
        onRowClick={(user) => console.log('Clicked:', user)}
        onSelectionChange={(ids) => setSelectedUsers(ids)}
        emptyState={{
          icon: 'people',
          title: 'No Users Found',
          description: 'Try adjusting your search or filters.'
        }}
      />
      
      // Custom column with rendering
      const columns = [
        {
          key: 'status',
          header: 'Status',
          width: 140,
          sortable: true,
          textWrap: 'nowrap',
          render: (value: string, row: User) => (
            <Badge variant={value === 'active' ? 'success' : 'warning'}>
              {value}
            </Badge>
          )
        }
      ];
    `,
    variants: ['default', 'bordered', 'striped', 'hoverable', 'compact', 'elevated', 'minimal'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    colors: ['primary', 'success', 'warning', 'danger', 'info', 'neutral'],
    props: {
      // Core Props
      data: 'T[] - Array of data objects',
      columns: 'ColumnConfig[] - Column configuration array',
      height: 'number - Fixed height for virtualization',
      rowHeight: 'number - Height of each row',
      
      // Visual Props
      size: 'VirtualizedTableSize - xs | sm | md | lg | xl',
      variant: 'VirtualizedTableVariant - default | bordered | striped | hoverable | compact | elevated | minimal',
      color: 'VirtualizedTableColor - primary | success | warning | danger | info | neutral',
      
      // Feature Props
      searchable: 'boolean - Enable search functionality',
      filterable: 'boolean - Enable filter functionality',
      selectable: 'boolean - Enable row selection',
      sortable: 'boolean - Enable sorting',
      multiSort: 'boolean - Enable multi-column sorting',
      striped: 'boolean - Alternating row colors',
      hoverable: 'boolean - Hover effects on rows',
      
      // Layout Props
      showHeader: 'boolean - Show table header',
      showFooter: 'boolean - Show table footer',
      showRowNumbers: 'boolean - Show row numbers',
      bordered: 'boolean - Add borders',
      hasBackground: 'boolean - Add background',
      isRounded: 'boolean - Rounded corners',
      hasShadow: 'boolean - Add shadow',
      isResponsive: 'boolean - Responsive design',
      isFullWidth: 'boolean - Full width',
      
      // Event Props
      onRowClick: '(row: T) => void - Row click handler',
      onSelectionChange: '(selectedIds: string[]) => void - Selection change handler',
      onSortChange: '(sortConfig: SortConfig[]) => void - Sort change handler',
      onFilterChange: '(filterKey: string, value: string) => void - Filter change handler',
      onError: '(error: string) => void - Error handler',
      
      // State Props
      loading: 'boolean - Loading state',
      highlightedRows: '(string | number)[] - Highlighted row IDs',
      highlightedColumns: 'string[] - Highlighted column keys',
      highlightColor: 'HighlightColor - Highlight color theme',
      
      // Customization Props
      className: 'string - Custom CSS classes',
      containerClassName: 'string - Container CSS classes',
      headerClassName: 'string - Header CSS classes',
      bodyClassName: 'string - Body CSS classes',
      footerClassName: 'string - Footer CSS classes',
      style: 'React.CSSProperties - Inline styles',
      
      // Content Props
      title: 'string - Table title',
      description: 'string - Table description',
      searchPlaceholder: 'string - Search input placeholder',
      emptyState: 'EmptyStateConfig - Empty state configuration'
    },
    performance: {
      virtualization: 'Handles 10,000+ rows smoothly',
      memoryUsage: 'Constant memory regardless of data size',
      rendering: 'Only visible rows are rendered',
      scrolling: '60fps smooth scrolling',
      searchFilter: 'Real-time performance'
    }
  },

  Footer: {
    title: 'Footer Component',
    description: 'Comprehensive footer component with compound structure, theme support, and industry-standard features',
    usage: `
      import { Footer } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic Footer
      <Footer>
        <Footer.Container>
          <div className="text-center">
            <Footer.Copyright company="My Company" />
          </div>
        </Footer.Container>
      </Footer>
      
      // Extended Footer with Theme
      <Footer theme="brand" variant="extended">
        <Footer.Container>
          <div className="grid grid-cols-1 md:grid-cols-4 gap-8">
            <Footer.Brand text="Acutrack" />
            <Footer.Section title="Quick Links">
              <Footer.Links links={links} />
            </Footer.Section>
            <Footer.Section title="Contact">
              <Footer.Contact email="hello@acutrack.com" />
            </Footer.Section>
            <Footer.Newsletter />
          </div>
          <div className="mt-8 pt-8 border-t">
            <Footer.Copyright />
            <Footer.Social socialLinks={socialLinks} />
          </div>
        </Footer.Container>
      </Footer>
    `,
    variants: ['default', 'minimal', 'extended', 'landing', 'dashboard'],
    sizes: ['sm', 'md', 'lg', 'xl'],
    themes: ['default', 'brand', 'minimal', 'grey', 'dark', 'custom'],
    subComponents: [
      'Footer.Container',
      'Footer.Section', 
      'Footer.Brand',
      'Footer.Links',
      'Footer.Social',
      'Footer.Contact',
      'Footer.Newsletter',
      'Footer.Copyright'
    ],
    props: {
      variant: 'Footer variant (default, minimal, extended, landing, dashboard)',
      size: 'Footer size (sm, md, lg, xl)',
      theme: 'Theme variant (default, brand, minimal, grey, dark, custom)',
      customColors: 'Custom color configuration object',
      sticky: 'Whether footer is sticky to bottom',
      className: 'Additional CSS classes'
    },
    features: [
      '5 Theme Variants with smart logo integration',
      'Compound structure with modular sub-components',
      'Industry standards: newsletter, social media, contact info',
      'Responsive design with mobile optimization',
      'WCAG 2.1 accessibility compliance',
      'Performance optimized with event-based updates'
    ]
  }

};

// Quick reference for AI assistants
export const AI_QUICK_REFERENCE = {
  // Always use these components
  components: ['Badge', 'Icon', 'Alert', 'Button', 'Header', 'Footer', 'Sidebar', 'SearchBar', 'Card', 'Grid', 'Layout', 'Container', 'VirtualizedTable', 'Select', 'Autocomplete', 'DatePicker', 'Toast'],
  
  // Never use these
  avoid: ['React Native components', 'External UI libraries', 'Hardcoded HTML elements'],
  
  // Design system colors
  colors: {
    primary: 'brand-orange',
    secondary: 'brand-navy', 
    success: 'system-success',
    warning: 'system-warning',
    error: 'system-error',
    info: 'system-info'
  },
  
  // Icon usage rules
  iconRules: [
    'Always use size prop: <Icon size="lg"> not className="w-6 h-6"',
    'Use aria-label for interactive icons',
    'Use ariaHidden={true} for decorative icons',
    'Require full range Material Symbols font for filled/weight support'
  ],
  
  // Badge usage rules
  badgeRules: [
    'Use appropriate variants for context (success, warning, error, info)',
    'Use isInteractive for clickable badges',
    'Use isDismissible for temporary notifications',
    'Use hasIcon with Icon component for enhanced meaning'
  ],
  
  // Layout usage rules
  layoutRules: [
    'Use Layout as the main page wrapper with hasMinHeight and isColumn',
    'Use Container for content width management with appropriate size variants',
    'Use Grid with responsive cols configuration for content organization',
    'Use Card for individual content blocks with appropriate variants',
    'Use LayoutHeader, Main, Footer for page structure',
    'Always use responsive column configurations: cols={{ xs: 1, sm: 2, md: 3 }}',
    'Use appropriate gap sizes (4-8 for most layouts)',
    'Use Container size variants based on content type (article, modal, etc.)'
  ],

  // Toast Component
  Toast: {
    title: 'Toast Component',
    description: 'Flexible toast notifications for user feedback with customizable variants, positioning, and timing',
    usage: `
      import { Toast, ToastProvider, useToast } from '@acutrack-bookprint/acutrack-ds';
      
      // Wrap your app with ToastProvider
      <ToastProvider defaultPosition="top-right" defaultSize="md">
        <App />
      </ToastProvider>
      
      // Use the hook to show toasts
      const { showToast } = useToast();
      
      showToast({
        title: 'Success!',
        description: 'Your action was completed successfully.',
        variant: 'success'
      });
    `,
    variants: ['success', 'error', 'info', 'warning', 'danger'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    positions: ['top-left', 'top-center', 'top-right', 'bottom-left', 'bottom-center', 'bottom-right'],
    features: [
      'Non-intrusive user feedback',
      'Customizable positioning and timing',
      'Auto-close functionality',
      'Manual dismissal support',
      'Dark mode support',
      'Accessibility features',
      'Design system integration'
    ]
  },

  // Toast usage rules
  toastRules: [
    'Use ToastProvider to wrap your app or component tree',
    'Use useToast hook to show toast notifications',
    'Use appropriate variants: success for completed actions, error for failures, warning for important info, info for updates',
    'Keep toast messages concise and clear',
    'Use autoClose for temporary messages, manual close for important ones',
    'Position toasts where they won\'t interfere with user actions',
    'Don\'t use toasts for critical errors that require immediate attention',
    'Don\'t stack too many toasts simultaneously',
    'Test toast behavior in both light and dark modes'
  ],

  // Dropdown Components
  Select: {
    title: 'Select Component',
    description: 'Dropdown select component for choosing from predefined options with overflow protection and design system integration',
    usage: `
      import { Select } from '@acutrack-bookprint/acutrack-ds';
      
      const options = [
        { value: 'option1', label: 'Option 1' },
        { value: 'option2', label: 'Option 2' }
      ];
      
      <Select
        label="Choose an option"
        id="my-select"
        options={options}
        value={selectedValue}
        onChange={setSelectedValue}
        isClearable={true}
        showDescriptions={true}
      />
    `,
    variants: ['primary', 'secondary', 'outline', 'ghost', 'filled'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    colors: ['primary', 'success', 'warning', 'danger', 'info'],
    features: [
      'Dropdown overflow protection (z-index: 9999)',
      'Design system color integration',
      'Dark mode support',
      'Clear button functionality',
      'Option descriptions',
      'Error states',
      'Disabled states',
      'Keyboard navigation'
    ]
  },

  Autocomplete: {
    title: 'Autocomplete Component',
    description: 'Type-ahead search component with filtering, suggestions, and create options',
    usage: `
      import { Autocomplete } from '@acutrack-bookprint/acutrack-ds';
      
      <Autocomplete
        label="Search fruits"
        id="fruit-search"
        options={fruitOptions}
        value={selectedValue}
        onChange={setSelectedValue}
        showCreateOption={true}
        showDescriptions={true}
        placeholder="Type to search..."
      />
    `,
    variants: ['primary', 'secondary', 'outline', 'ghost', 'filled'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    colors: ['primary', 'success', 'warning', 'danger', 'info'],
    features: [
      'Real-time filtering',
      'Create new options',
      'Option icons and descriptions',
      'Keyboard navigation',
      'Dropdown overflow protection',
      'Design system integration',
      'Dark mode support'
    ]
  },

  DatePicker: {
    title: 'DatePicker Component',
    description: 'Date selection component with calendar popover and date constraints',
    usage: `
      import { DatePicker } from '@acutrack-bookprint/acutrack-ds';
      
      <DatePicker
        label="Select a date"
        id="my-datepicker"
        value={selectedDate}
        onChange={setSelectedDate}
        minDate={new Date()}
        maxDate={new Date(Date.now() + 30 * 24 * 60 * 60 * 1000)}
        hasClearButton={true}
        hasTodayButton={true}
      />
    `,
    variants: ['primary', 'secondary', 'outline', 'ghost', 'text'],
    sizes: ['xs', 'sm', 'md', 'lg', 'xl'],
    colors: ['primary', 'secondary', 'success', 'warning', 'danger', 'info', 'neutral'],
    features: [
      'Calendar popover or inline mode',
      'Date constraints (min/max)',
      'Clear and today buttons',
      'Popover overflow protection',
      'Design system integration',
      'Dark mode support',
      'Keyboard navigation'
    ]
  },

  // Dropdown usage rules
  dropdownRules: [
    'All dropdown components now have overflow protection with z-index: 9999',
    'Use appropriate size variants based on your layout context',
    'Use clear buttons for optional selections',
    'Use option descriptions for complex choices',
    'Test dropdown behavior in containers with overflow: hidden',
    'Use design system colors - no hardcoded values',
    'Ensure proper keyboard navigation support',
    'Test in both light and dark modes',
    'Use appropriate placeholder text',
    'Consider using create options for dynamic data entry'
  ],

  // Button usage rules
  buttonRules: [
    'Use primary variant for main call-to-action buttons',
    'Use secondary variant for secondary actions',
    'Use text or ghost variants for subtle actions',
    'Use danger variant sparingly for destructive actions',
              'Hover effects are enabled by default for better UX - disable only when needed',
    'Provide clear, actionable text in button labels',
    'Use icons to enhance button meaning when appropriate',
    'Use loading states for async operations',
    'Disable buttons when actions are not available',
    'Test button accessibility with keyboard navigation',
    'Use appropriate sizes based on context and importance'
  ],

  // Footer usage rules
  footerRules: [
    'Use Footer.Container to wrap all footer content',
    'Use Footer.Section to group related links and content',
    'Use Footer.Brand for logo and company branding with smart theme-aware switching',
    'Use Footer.Links for navigation links with proper structure',
    'Use Footer.Social for social media links with icons',
    'Use Footer.Contact for contact information display',
    'Use Footer.Newsletter for email subscription forms',
    'Use Footer.Copyright for copyright text with year and company',
    'Choose appropriate theme: default for design system colors, brand for navy/orange, minimal for clean white',
    'Use responsive grid layouts: grid-cols-1 md:grid-cols-2 lg:grid-cols-4',
    'Group related links into logical sections with clear titles',
    'Limit each section to 5-7 links maximum for better UX',
    'Test footer in both light and dark modes',
    'Ensure proper mobile responsiveness with stacked layouts',
    'Use semantic HTML structure for accessibility',
    'Provide proper ARIA labels for screen readers'
  ]
};
