// @acutrack-bookprint/acutrack-ds/dist/docs/api/LayoutDocumentation.ts
const LayoutDocumentation = {
  title: "Layout Components",
  description: "Comprehensive layout system for building responsive, screen-friendly interfaces",
  category: "Layout",
  version: "1.0.0",
  
  overview: {
    description: "The Acutrack Design System provides a comprehensive set of layout components to help you build responsive, screen-friendly layouts. These components work together to create consistent, accessible, and maintainable user interfaces.",
    keyFeatures: [
      "Responsive grid system with flexible column configurations",
      "Multiple container sizes and variants for content management",
      "Flexible card system with interactive states and variants",
      "Complete layout structure components (header, main, sidebar, footer)",
      "Multi-column layout helpers for complex page structures",
      "Built-in accessibility features and semantic HTML",
      "Consistent spacing and design system integration"
    ]
  },

  components: {
    Layout: {
      description: "Main layout wrapper that provides the foundation for your page structure",
      props: {
        children: "React.ReactNode - Content to display inside the layout",
        size: "'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | 'full' | 'screen' - Layout size variant",
        variant: "'default' | 'dashboard' | 'landing' | 'admin' | 'minimal' | 'elevated' - Visual variant",
        color: "'primary' | 'success' | 'warning' | 'danger' | 'info' - Color theme",
        hasMinHeight: "boolean (default: true) - Whether layout has minimum viewport height",
        isColumn: "boolean (default: true) - Whether layout uses flexbox column direction",
        isFullWidth: "boolean (default: true) - Whether layout is full width",
        hasBackground: "boolean (default: false) - Whether layout has background",
        hasBorder: "boolean (default: false) - Whether layout has border",
        hasRoundedCorners: "boolean (default: false) - Whether layout has rounded corners",
        hasShadow: "boolean (default: false) - Whether layout has shadow",
        isLoading: "boolean (default: false) - Whether layout is in loading state",
        isError: "boolean (default: false) - Whether layout is in error state",
        isWarning: "boolean (default: false) - Whether layout is in warning state",
        isSuccess: "boolean (default: false) - Whether layout is in success state",
        isInfo: "boolean (default: false) - Whether layout is in info state"
      }
    },

    Container: {
      description: "Manages content width and provides responsive containers with various size options",
      props: {
        size: "'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | 'full' | 'screen' (default: 'screen') - Container size variant",
        variant: "'default' | 'article' | 'main' | 'aside' | 'section' | 'card' | 'panel' | 'modal' (default: 'default') - Visual variant",
        padding: "'none' | 'sm' | 'md' | 'lg' | 'xl' (default: 'md') - Padding size",
        isCentered: "boolean (default: true) - Whether container is centered horizontally",
        isFullWidth: "boolean (default: false) - Whether container is full width",
        hasBackground: "boolean (default: false) - Whether container has background",
        hasBorder: "boolean (default: false) - Whether container has border",
        hasRoundedCorners: "boolean (default: false) - Whether container has rounded corners",
        hasShadow: "boolean (default: false) - Whether container has shadow"
      },
      sizeVariants: {
        xs: "max-w-xs (320px) - Small forms, modals",
        sm: "max-w-sm (384px) - Cards, small content",
        md: "max-w-md (448px) - Medium content blocks",
        lg: "max-w-lg (512px) - Articles, forms",
        xl: "max-w-xl (576px) - Large content blocks",
        "2xl": "max-w-2xl (672px) - Wide content blocks",
        full: "max-w-full - Full width content",
        screen: "max-w-screen-lg - Screen-constrained content"
      }
    },

    Grid: {
      description: "Provides a responsive grid system for organizing content into columns and rows",
      props: {
        cols: "number | ResponsiveColumns (default: 1) - Number of columns or responsive config",
        gap: "1 | 2 | 3 | 4 | 5 | 6 | 8 | 10 | 12 | 16 | 20 | 24 | 32 (default: 6) - Gap size between grid items",
        hasBackground: "boolean (default: false) - Whether grid has background",
        isBordered: "boolean (default: false) - Whether grid has borders",
        isRounded: "boolean (default: false) - Whether grid has rounded corners",
        hasShadow: "boolean (default: false) - Whether grid has shadow",
        hasBackdropBlur: "boolean (default: false) - Whether grid has backdrop blur",
        hasGlassMorphism: "boolean (default: false) - Whether grid has glass morphism effect",
        isLoading: "boolean (default: false) - Whether grid is in loading state",
        hasLoadingText: "boolean (default: false) - Whether to show loading text",
        loadingText: "string (default: 'Loading...') - Text to display during loading"
      },
      responsiveColumns: {
        xs: "Number of columns on extra small screens (default)",
        sm: "Number of columns on small screens and up",
        md: "Number of columns on medium screens and up",
        lg: "Number of columns on large screens and up",
        xl: "Number of columns on extra large screens and up",
        "2xl": "Number of columns on 2xl screens and up"
      }
    },

    Card: {
      description: "Flexible content containers with multiple variants and interactive states",
      props: {
        size: "'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' (default: 'md') - Card size variant",
        variant: "'default' | 'minimal' | 'elevated' | 'bordered' | 'filled' | 'outlined' | 'ghost' | 'card' | 'panel' (default: 'default') - Visual variant",
        color: "'primary' | 'success' | 'warning' | 'danger' | 'info' | 'neutral' (default: 'primary') - Color theme",
        isInteractive: "boolean (default: false) - Whether card has hover effects",
        isFeatured: "boolean (default: false) - Whether card is featured",
        featuredText: "string (default: 'Featured') - Text for featured badge",
        isBordered: "boolean (default: true) - Whether card has border",
        hasShadow: "boolean (default: true) - Whether card has shadow",
        isRounded: "boolean (default: true) - Whether card has rounded corners",
        fullWidth: "boolean (default: false) - Whether card takes full width",
        hasPadding: "boolean (default: true) - Whether card has padding",
        disabled: "boolean (default: false) - Whether card is disabled"
      },
      subComponents: {
        CardImage: "Image component with aspect ratio and sizing options",
        CardHeader: "Header section with optional border and padding",
        CardBody: "Main content area with scrollable and height options",
        CardFooter: "Footer section with alignment options"
      }
    }
  },

  layoutStructure: {
    LayoutHeader: "Page header with sticky positioning and elevation options",
    Main: "Main content area with responsive width management",
    LayoutSidebar: "Responsive sidebar with configurable width and positioning",
    Footer: "Page footer with elevation and background options",
    Content: "Content wrapper with padding and styling options"
  },

  multiColumnLayouts: {
    TwoColumnLayout: "Two-column responsive layout with configurable sidebar position",
    ThreeColumnLayout: "Three-column responsive layout"
  },

  usage: {
    basicLayout: `
import { Layout, LayoutHeader, Main, Footer } from '@acutrack-bookprint/acutrack-ds';

function App() {
  return (
    <Layout hasMinHeight isColumn isFullWidth>
      <LayoutHeader isSticky hasElevation>
        <h1>Page Title</h1>
      </LayoutHeader>
      <Main hasPadding>
        <div>Main content</div>
      </Main>
      <Footer hasBackground>
        <p>© 2024 Acutrack</p>
      </Footer>
    </Layout>
  );
}`,

    responsiveGrid: `
import { Grid, Card } from '@acutrack-bookprint/acutrack-ds';

function ProductGrid() {
  return (
    <Grid cols={{ xs: 1, sm: 2, md: 3, lg: 4 }} gap={6}>
      {products.map(product => (
        <Card key={product.id} isInteractive>
          <Card.Image src={product.image} alt={product.name} />
          <Card.Body>
            <h3>{product.name}</h3>
            <p>{product.price}</p>
          </Card.Body>
        </Card>
      ))}
    </Grid>
  );
}`,

    containerSizes: `
import { Container } from '@acutrack-bookprint/acutrack-ds';

function ContentSections() {
  return (
    <>
      <Container size="xs" isCentered>
        <p>Small form or modal content</p>
      </Container>
      
      <Container size="lg" variant="article" hasBackground>
        <h1>Article Content</h1>
        <p>Article body text...</p>
      </Container>
      
      <Container size="screen" padding="lg">
        <p>Full-width section content</p>
      </Container>
    </>
  );
}`
  },

  bestPractices: {
    responsiveDesign: [
      "Always use responsive column configurations for Grid components",
      "Test layouts on different screen sizes",
      "Use Container size variants appropriately for content width"
    ],
    accessibility: [
      "Use semantic HTML elements when possible",
      "Ensure proper heading hierarchy",
      "Provide adequate color contrast",
      "Include proper ARIA labels for interactive elements"
    ],
    performance: [
      "Use appropriate Card variants to avoid unnecessary styling",
      "Implement lazy loading for large grids",
      "Optimize images in Card.Image components"
    ],
    consistency: [
      "Stick to the design system's spacing scale",
      "Use consistent gap sizes across similar layouts",
      "Maintain consistent padding and margins"
    ],
    contentOrganization: [
      "Use Layout structure components for page-level organization",
      "Use Grid for content organization within sections",
      "Use Card for individual content blocks",
      "Use Container for content width management"
    ]
  },

  troubleshooting: {
    commonIssues: {
      "Grid items not responsive": "Ensure you're using the responsive cols prop with an object and check that Tailwind CSS is properly configured",
      "Cards not displaying correctly": "Verify that you're using the correct Card sub-components and check that the Card variant and size props are valid",
      "Layout not full height": "Ensure hasMinHeight is set to true on the Layout component and check that parent containers have proper height settings",
      "Container not centering": "Verify that isCentered is set to true and check that the Container has a defined width"
    },
    cssImport: "Make sure to import the design system's CSS: import '@acutrack-bookprint/acutrack-ds/style.css';"
  },

      examples: {
      dashboard: "Complete dashboard layout with header, sidebar, main content, and footer",
      landingPage: "Landing page layout with hero section, features grid, and footer",
      productGrid: "Responsive product grid with cards and interactive elements",
      articleLayout: "Article layout with proper typography and responsive containers",
      formLayout: "Form layout with proper spacing and container management"
    }
  };

export default LayoutDocumentation;
