// @acutrack-bookprint/acutrack-ds/dist/docs/api/RadioDocumentation.ts
export default {
  title: 'Radio Component',
  description: 'Single-choice form input with multiple variants, sizes, and states for user selection',
  
  // Basic usage examples
  basicUsage: `
    import { Radio } from '@acutrack-bookprint/acutrack-ds';
    
    // Basic radio button
    <Radio 
      id="option1"
      name="choice"
      value="option1"
      label="Option 1"
    />
    
    // Radio group
    <div>
      <Radio id="small" name="size" value="small" label="Small" />
      <Radio id="medium" name="size" value="medium" label="Medium" />
      <Radio id="large" name="size" value="large" label="Large" />
    </div>
    
    // With description
    <Radio 
      id="premium"
      name="plan"
      value="premium"
      label="Premium Plan"
      description="Includes all features and priority support"
    />
  `,
  
  // Advanced usage examples
  advancedUsage: `
    import { Radio } from '@acutrack-bookprint/acutrack-ds';
    
    // Different variants
    <Radio variant="success" id="success" name="status" value="success" label="Success" />
    <Radio variant="warning" id="warning" name="status" value="warning" label="Warning" />
    <Radio variant="error" id="error" name="status" value="error" label="Error" />
    
    // Different sizes
    <Radio size="sm" id="small" name="size" value="small" label="Small" />
    <Radio size="md" id="medium" name="size" value="medium" label="Medium" />
    <Radio size="lg" id="large" name="size" value="large" label="Large" />
    
    // With state management
    const [selectedValue, setSelectedValue] = useState('');
    
    <Radio 
      id="option1"
      name="choice"
      value="option1"
      label="Option 1"
      checked={selectedValue === 'option1'}
      onChange={(e) => setSelectedValue(e.target.value)}
    />
    
    // Disabled state
    <Radio 
      id="disabled"
      name="choice"
      value="disabled"
      label="Disabled Option"
      isDisabled
    />
    
    // Loading state
    <Radio 
      id="loading"
      name="choice"
      value="loading"
      label="Loading Option"
      isLoading
    />
    
    // Required field
    <Radio 
      id="required"
      name="choice"
      value="required"
      label="Required Option"
      isRequired
      showRequiredIndicator
    />
  `,
  
  // Available variants
  variants: {
    primary: 'Primary brand color (orange theme)',
    success: 'Success state (green theme)',
    warning: 'Warning state (yellow theme)',
    danger: 'Error/danger state (red theme)',
    info: 'Information state (blue theme)'
  },
  
  // Available sizes
  sizes: {
    sm: 'Small radio buttons (14px)',
    md: 'Default size (16px)',
    lg: 'Large radio buttons (18px)',
    xl: 'Extra large radio buttons (20px)'
  },
  
  // Key props
  keyProps: {
    id: 'Unique identifier for the radio button (required)',
    name: 'Name attribute for grouping radio buttons (required)',
    value: 'Value of the radio button (required)',
    label: 'Label text for the radio button',
    description: 'Optional description text below the label',
    checked: 'Whether the radio button is selected',
    onChange: 'Callback function when selection changes',
    variant: 'Visual variant (primary, success, warning, danger, info)',
    size: 'Size of the radio button (sm, md, lg, xl)',
    isDisabled: 'Whether the radio button is disabled',
    isLoading: 'Whether the radio button is in loading state',
    isRequired: 'Whether the radio button is required',
    showRequiredIndicator: 'Whether to show asterisk for required fields',
    className: 'Additional CSS classes',
    labelClassName: 'Additional CSS classes for the label'
  },
  
  // Best practices
  bestPractices: {
    do: [
      'Always provide unique id and name attributes',
      'Use descriptive labels that clearly indicate the choice',
      'Group related radio buttons with the same name attribute',
      'Use appropriate variants for different contexts',
      'Provide descriptions for complex options',
      'Use required indicators for mandatory fields',
      'Handle state changes with onChange callbacks'
    ],
    dont: [
      'Use radio buttons for multiple selections (use checkboxes instead)',
      'Use radio buttons for actions (use buttons instead)',
      'Make labels too long or complex',
      'Use radio buttons without proper grouping',
      'Forget to handle the onChange event',
      'Use radio buttons for binary choices when a toggle might be better'
    ]
  },
  
  // Form integration
  formIntegration: {
    controlled: `
      const [value, setValue] = useState('');
      
      <Radio 
        id="option1"
        name="choice"
        value="option1"
        label="Option 1"
        checked={value === 'option1'}
        onChange={(e) => setValue(e.target.value)}
      />
    `,
    uncontrolled: `
      <form onSubmit={handleSubmit}>
        <Radio id="option1" name="choice" value="option1" label="Option 1" />
        <Radio id="option2" name="choice" value="option2" label="Option 2" />
        <button type="submit">Submit</button>
      </form>
    `,
    withFormLibrary: `
      // With React Hook Form
      <Radio 
        id="option1"
        name="choice"
        value="option1"
        label="Option 1"
        {...register('choice')}
      />
    `
  },
  
  // Design system integration
  designSystem: {
    colors: 'Uses design system colors (brand-orange, system-success, system-warning, system-error, system-info)',
    darkMode: 'Automatically supports dark mode with proper color variants',
    accessibility: 'Includes proper ARIA attributes, keyboard navigation, and screen reader support',
    consistency: 'Follows design system spacing, typography, and interaction patterns',
    forms: 'Integrates with Tailwind CSS forms plugin for consistent styling'
  },
  
  // Common use cases
  useCases: {
    settings: `
      <div>
        <h3>Theme Preference</h3>
        <Radio id="light" name="theme" value="light" label="Light Theme" />
        <Radio id="dark" name="theme" value="dark" label="Dark Theme" />
        <Radio id="auto" name="theme" value="auto" label="Auto (System)" />
      </div>
    `,
    preferences: `
      <div>
        <h3>Notification Settings</h3>
        <Radio id="all" name="notifications" value="all" label="All Notifications" />
        <Radio id="important" name="notifications" value="important" label="Important Only" />
        <Radio id="none" name="notifications" value="none" label="No Notifications" />
      </div>
    `,
    selection: `
      <div>
        <h3>Choose Plan</h3>
        <Radio 
          id="basic"
          name="plan"
          value="basic"
          label="Basic Plan"
          description="Essential features for individuals"
        />
        <Radio 
          id="pro"
          name="plan"
          value="pro"
          label="Pro Plan"
          description="Advanced features for professionals"
        />
        <Radio 
          id="enterprise"
          name="plan"
          value="enterprise"
          label="Enterprise Plan"
          description="Full features for organizations"
        />
      </div>
    `
  },
  
  // Accessibility features
  accessibility: {
    ariaLabels: 'Proper label association with htmlFor attribute',
    keyboardNavigation: 'Full keyboard support with Tab and Space/Enter',
    screenReader: 'Proper ARIA attributes and descriptions',
    focusManagement: 'Clear focus indicators and management',
    colorContrast: 'Meets WCAG contrast requirements'
  },
  
  // Troubleshooting
  troubleshooting: {
    notSelectable: [
      'Ensure the radio button has proper id and name attributes',
      'Check that the label is properly associated with htmlFor',
      'Verify that the radio button is not disabled',
      'Make sure the onChange handler is properly set up'
    ],
    colorsNotAppearing: [
      'Ensure you\'re using design system variants',
      'Check that dark mode classes are properly applied',
      'Verify that custom colors are valid CSS values'
    ],
    notGrouping: [
      'Ensure all radio buttons in a group have the same name attribute',
      'Check that each radio button has a unique id',
      'Verify that the values are different for each option'
    ],
    stateNotUpdating: [
      'Ensure you\'re handling the onChange event',
      'Check that the state is properly managed',
      'Verify that the checked prop is correctly set'
    ]
  },
  
  // Size guidelines
  sizeGuidelines: {
    sm: 'Small radio buttons for compact forms (14px)',
    md: 'Default size for most use cases (16px)',
    lg: 'Large radio buttons for touch interfaces (18px)',
    xl: 'Extra large radio buttons for accessibility (20px)'
  },
  
  // Variant guidelines
  variantGuidelines: {
    primary: 'Default choice, standard selection',
    success: 'Positive choice, recommended option',
    warning: 'Caution choice, important consideration',
    danger: 'Critical choice, warning option',
    info: 'Informational choice, additional option'
  }
};
