// @acutrack-bookprint/acutrack-ds/dist/docs/api/ThemeDocumentation.ts
export default {
  componentName: 'Theme System',
  description: 'Comprehensive dark and light mode theme implementation for the Acutrack Design System using Tailwind CSS class-based dark mode.',
  
  purpose: 'The theme system provides seamless switching between light and dark modes with automatic color adaptation, system preference detection, and persistent user preferences across all components.',
  
  overview: {
    implementation: 'Class-based dark mode using Tailwind CSS',
    strategy: 'CSS custom properties with Tailwind dark: prefix',
    persistence: 'localStorage with system preference fallback',
    detection: 'Automatic detection of user system preferences',
    components: 'All components support both light and dark themes'
  },
  
  setup: {
    tailwindConfig: `
      // tailwind.config.js
      export default {
        content: ["./src/**/*.{js,ts,jsx,tsx}"],
        darkMode: 'class', // Enable class-based dark mode
        theme: {
          extend: {
            colors: {
              // Light mode colors
              'text-primary': '#1B3A7B',
              'text-secondary': '#3C4858',
              'bg-primary': '#f6f7fa',
              'bg-secondary': '#FFFFFF',
              'bg-tertiary': '#ebeff9',
              'border-color': '#DFE4E9',
              
              // Dark mode colors
              'dark-text-primary': '#F6F8FD',
              'dark-text-secondary': '#9CA3AF',
              'dark-bg-primary': '#1c212c',
              'dark-bg-secondary': '#101620',
              'dark-bg-tertiary': '#1e2a3a',
              'dark-border-color': '#1c3332',
            }
          }
        }
      }
    `,
    
    htmlSetup: `
      <!-- Add dark class to html element for dark mode -->
      <html class="dark">
        <!-- Your app content -->
      </html>
    `,
    
    cssSetup: `
      /* Ensure smooth transitions between themes */
      * {
        transition: background-color 0.3s ease, color 0.3s ease, border-color 0.3s ease;
      }
    `
  },
  
  themeToggle: {
    description: 'ThemeToggle component for switching between light and dark modes',
    features: [
      'Automatic system preference detection',
      'localStorage persistence',
      'Smooth transitions between themes',
      'Accessible with proper ARIA labels',
      'Icon and text indicators'
    ],
    
    usage: `
      import { ThemeToggle } from '@acutrack-bookprint/acutrack-ds';
      
      // Basic usage
      <ThemeToggle />
      
      // With custom styling
      <ThemeToggle className="custom-theme-toggle" />
    `,
    
    implementation: `
      import React, { useState, useEffect } from 'react';
      import { Button } from './Button';
      import { Icon } from './icons';
      
      export const ThemeToggle: React.FC<{ className?: string }> = ({ className = "" }) => {
        const [theme, setTheme] = useState<'light' | 'dark'>('light');
      
        // Initialize theme from localStorage and system preference
        useEffect(() => {
          const storedTheme = localStorage.getItem('theme') as 'light' | 'dark' | null;
          if (storedTheme && (storedTheme === 'light' || storedTheme === 'dark')) {
            setTheme(storedTheme);
          } else {
            const prefersDark = window.matchMedia && 
              window.matchMedia('(prefers-color-scheme: dark)').matches;
            setTheme(prefersDark ? 'dark' : 'light');
          }
        }, []);
      
        // Apply theme to document
        const applyTheme = (themeToApply: 'light' | 'dark') => {
          const html = document.documentElement;
          html.classList.toggle('dark', themeToApply === 'dark');
          html.classList.toggle('light', themeToApply === 'light');
        };
      
        // Theme toggle handler
        const handleThemeToggle = () => {
          const newTheme = theme === 'light' ? 'dark' : 'light';
          setTheme(newTheme);
          applyTheme(newTheme);
          localStorage.setItem('theme', newTheme);
        };
      
        // Apply theme when it changes
        useEffect(() => {
          applyTheme(theme);
        }, [theme]);
      
        return (
          <Button
            variant="text"
            size="sm"
            onClick={handleThemeToggle}
            className={\`p-2 text-text-primary dark:text-dark-text-primary hover:bg-bg-secondary dark:hover:bg-dark-bg-secondary \${className}\`}
            aria-label={\`Switch to \${theme === 'light' ? 'dark' : 'light'} theme\`}
          >
            <div className="flex items-center space-x-2">
              <Icon className="w-5 h-5">
                {theme === 'light' ? 'dark_mode' : 'light_mode'}
              </Icon>
              <span className="text-sm font-medium capitalize">
                {theme === 'light' ? 'Light' : 'Dark'}
              </span>
            </div>
          </Button>
        );
      };
    `
  },
  
  colorSystem: {
    lightMode: {
      text: {
        primary: '#1B3A7B', // brand-navy
        secondary: '#3C4858', // neutral-600
        tertiary: '#6B7280', // neutral-500
        onAccent: '#FFFFFF'
      },
      background: {
        primary: '#f6f7fa', // light grey
        secondary: '#FFFFFF', // white
        tertiary: '#ebeff9' // very light blue-grey
      },
      border: {
        color: '#DFE4E9', // light grey
        strong: '#5fead4' // accent color
      },
      brand: {
        navy: '#1B3A7B',
        orange: '#0d9488',
        orangeHover: '#0a6960'
      }
    },
    
    darkMode: {
      text: {
        primary: '#F6F8FD', // very light blue
        secondary: '#9CA3AF', // neutral-400
        tertiary: '#6B7280', // neutral-500
        onAccent: '#FFFFFF'
      },
      background: {
        primary: '#1c212c', // dark blue-grey
        secondary: '#101620', // darker blue-grey
        tertiary: '#1e2a3a' // medium blue-grey
      },
      border: {
        color: '#1c3332', // dark teal
        strong: '#ffffff' // white
      },
      brand: {
        navy: '#53e3cb', // light teal
        orange: '#0d9488', // same as light
        orangeHover: '#5eead4' // lighter teal
      }
    }
  },
  
  implementation: {
    basicUsage: `
      // Basic component with theme support
      <div className="bg-bg-primary dark:bg-dark-bg-primary text-text-primary dark:text-dark-text-primary">
        <h1 className="text-2xl font-bold">My Component</h1>
        <p className="text-text-secondary dark:text-dark-text-secondary">
          This text adapts to the current theme
        </p>
      </div>
    `,
    
    componentExample: `
      // Card component with full theme support
      const Card = ({ children, className = "" }) => (
        <div className={\`
          bg-bg-secondary dark:bg-dark-bg-secondary
          border border-border-color dark:border-dark-border-color
          text-text-primary dark:text-dark-text-primary
          rounded-lg p-6 shadow-sm
          \${className}
        \`}>
          {children}
        </div>
      );
    `,
    
    buttonExample: `
      // Button with theme-aware colors
      const Button = ({ variant = "primary", children, ...props }) => {
        const baseClasses = "px-4 py-2 rounded font-medium transition-colors";
        
        const variantClasses = {
          primary: "bg-brand-orange text-white hover:bg-brand-orange-hover dark:bg-dark-brand-orange dark:hover:bg-dark-brand-orange-hover",
          secondary: "bg-bg-tertiary text-brand-navy hover:bg-bg-secondary dark:bg-dark-bg-tertiary dark:text-dark-brand-navy dark:hover:bg-dark-bg-secondary",
          outline: "border border-brand-navy text-brand-navy hover:bg-brand-navy hover:text-white dark:border-dark-brand-navy dark:text-dark-brand-navy dark:hover:bg-dark-brand-navy dark:hover:text-white"
        };
        
        return (
          <button className={\`\${baseClasses} \${variantClasses[variant]}\`} {...props}>
            {children}
          </button>
        );
      };
    `,
    
    formExample: `
      // Form input with theme support
      const Input = ({ className = "", ...props }) => (
        <input
          className={\`
            w-full px-3 py-2
            bg-bg-secondary dark:bg-dark-bg-secondary
            border border-border-color dark:border-dark-border-color
            text-text-primary dark:text-dark-text-primary
            placeholder:text-text-secondary dark:placeholder:text-dark-text-secondary
            focus:outline-none focus:ring-2 focus:ring-brand-orange
            rounded-md
            \${className}
          \`}
          {...props}
        />
      );
    `
  },
  
  bestPractices: [
    'Always provide both light and dark variants for every color',
    'Use semantic color names (text-primary, bg-secondary) instead of specific colors',
    'Test contrast ratios in both light and dark modes',
    'Provide smooth transitions between theme changes',
    'Use the dark: prefix consistently for dark mode styles',
    'Group related theme styles together for better readability',
    'Test with system preference changes (auto theme switching)',
    'Ensure all interactive elements have proper focus states in both themes',
    'Use consistent spacing and typography across themes',
    'Test accessibility with screen readers in both modes'
  ],
  
  accessibility: [
    'Maintain WCAG AA contrast ratios in both themes',
    'Provide clear visual indicators for theme state',
    'Support system preference detection',
    'Ensure keyboard navigation works in both themes',
    'Use proper ARIA labels for theme toggle',
    'Test with high contrast mode enabled',
    'Provide alternative ways to access theme toggle'
  ],
  
  testing: {
    manual: [
      'Test theme toggle functionality',
      'Verify localStorage persistence',
      'Test system preference detection',
      'Check all components in both themes',
      'Verify smooth transitions',
      'Test with different screen sizes'
    ],
    
    automated: `
      // Example test for theme toggle
      import { render, screen, fireEvent } from '@testing-library/react';
      import { ThemeToggle } from './ThemeToggle';
      
      test('toggles theme correctly', () => {
        render(<ThemeToggle />);
        
        const toggleButton = screen.getByRole('button');
        expect(toggleButton).toHaveAttribute('aria-label', 'Switch to dark theme');
        
        fireEvent.click(toggleButton);
        expect(toggleButton).toHaveAttribute('aria-label', 'Switch to light theme');
        expect(document.documentElement).toHaveClass('dark');
      });
    `
  },
  
  troubleshooting: {
    commonIssues: [
      {
        issue: 'Theme not persisting after page refresh',
        solution: 'Ensure localStorage is available and theme is saved on toggle'
      },
      {
        issue: 'System preference not detected',
        solution: 'Check if matchMedia is supported and handle gracefully'
      },
      {
        issue: 'Dark mode styles not applying',
        solution: 'Verify darkMode: "class" is set in Tailwind config and dark class is on html element'
      },
      {
        issue: 'Flickering during theme switch',
        solution: 'Apply theme before rendering components or use CSS transitions'
      },
      {
        issue: 'Some components not theme-aware',
        solution: 'Add dark: variants to all color-related classes'
      }
    ]
  },
  
  migration: {
    fromNoTheme: [
      'Add darkMode: "class" to Tailwind config',
      'Add dark: variants to all color classes',
      'Implement ThemeToggle component',
      'Add theme persistence logic',
      'Test all components in both themes'
    ],
    
    fromOtherThemeSystem: [
      'Replace existing theme logic with class-based approach',
      'Update all color references to use design system tokens',
      'Replace theme context with direct class manipulation',
      'Update component props to remove theme-specific variants'
    ]
  },
  
  examples: {
    completeApp: `
      // Complete app with theme support
      import React, { useState, useEffect } from 'react';
      import { ThemeToggle } from '@acutrack-bookprint/acutrack-ds';
      
      const App = () => {
        const [theme, setTheme] = useState<'light' | 'dark'>('light');
      
        useEffect(() => {
          const storedTheme = localStorage.getItem('theme') as 'light' | 'dark' | null;
          if (storedTheme) {
            setTheme(storedTheme);
          } else {
            const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
            setTheme(prefersDark ? 'dark' : 'light');
          }
        }, []);
      
        useEffect(() => {
          document.documentElement.classList.toggle('dark', theme === 'dark');
        }, [theme]);
      
        return (
          <div className="min-h-screen bg-bg-primary dark:bg-dark-bg-primary">
            <header className="bg-bg-secondary dark:bg-dark-bg-secondary border-b border-border-color dark:border-dark-border-color">
              <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
                <div className="flex justify-between items-center h-16">
                  <h1 className="text-xl font-semibold text-text-primary dark:text-dark-text-primary">
                    My App
                  </h1>
                  <ThemeToggle />
                </div>
              </div>
            </header>
            
            <main className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8">
              <div className="bg-bg-secondary dark:bg-dark-bg-secondary rounded-lg p-6">
                <h2 className="text-2xl font-bold text-text-primary dark:text-dark-text-primary mb-4">
                  Welcome to My App
                </h2>
                <p className="text-text-secondary dark:text-dark-text-secondary">
                  This app supports both light and dark themes with automatic system preference detection.
                </p>
              </div>
            </main>
          </div>
        );
      };
    `,
    
    componentLibrary: `
      // Theme-aware component library
      const Card = ({ children, className = "" }) => (
        <div className={\`
          bg-bg-secondary dark:bg-dark-bg-secondary
          border border-border-color dark:border-dark-border-color
          rounded-lg p-6 shadow-sm
          \${className}
        \`}>
          {children}
        </div>
      );
      
      const Button = ({ variant = "primary", children, ...props }) => {
        const variants = {
          primary: "bg-brand-orange text-white hover:bg-brand-orange-hover dark:bg-dark-brand-orange dark:hover:bg-dark-brand-orange-hover",
          secondary: "bg-bg-tertiary text-brand-navy hover:bg-bg-secondary dark:bg-dark-bg-tertiary dark:text-dark-brand-navy dark:hover:bg-dark-bg-secondary"
        };
        
        return (
          <button className={\`px-4 py-2 rounded font-medium transition-colors \${variants[variant]}\`} {...props}>
            {children}
          </button>
        );
      };
    `
  }
};
