// @acutrack-bookprint/acutrack-ds/dist/docs/api/ToastDocumentation.ts
// FILE: docs/ToastDocumentation.ts
// Comprehensive documentation for Toast component

export const ToastDocumentation = {
  name: 'Toast',
  description: 'A comprehensive toast notification system with flexible positioning, multiple variants, and context-based management.',
  category: 'Feedback',
  version: '1.0.68',
  
  // Component Overview
  overview: {
    purpose: 'Display temporary notifications, alerts, and feedback messages to users',
    useCases: [
      'Success confirmations after user actions',
      'Error messages when operations fail',
      'Warning notifications for important information',
      'Info messages for general updates',
      'Danger alerts for critical actions'
    ],
    benefits: [
      'Non-intrusive user feedback',
      'Consistent design system styling',
      'Accessible with proper ARIA attributes',
      'Customizable positioning and timing',
      'Dark mode support'
    ]
  },

  // Props Documentation
  props: {
    // Toast Component Props (Internal - used by ToastProvider)
    toast: {
      type: 'ToastMessage',
      required: true,
      description: 'The toast message object containing id, title, description, and variant',
      example: `{
  id: 'unique-id',
  title: 'Success!',
  description: 'Your action was completed successfully.',
  variant: 'success',
  size?: 'md'
}`
    },
    onDismiss: {
      type: 'function',
      required: true,
      description: 'Callback function called when the toast is dismissed',
      example: '(id: string) => void'
    },
    size: {
      type: "'xs' | 'sm' | 'md' | 'lg' | 'xl'",
      required: false,
      default: "'md'",
      description: 'The size of the toast component',
      options: {
        xs: 'Extra small toast with minimal padding (p-2, text-xs, min-w-64)',
        sm: 'Small toast for compact layouts (p-3, text-sm, min-w-72)',
        md: 'Medium toast (default) (p-4, text-base, min-w-80)',
        lg: 'Large toast for emphasis (p-5, text-lg, min-w-96)',
        xl: 'Extra large toast for important messages (p-6, text-xl, min-w-[28rem])'
      }
    },
    hasIcon: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Whether to show an icon in the toast'
    },
    hasCloseButton: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Whether to show a close button on the toast'
    },
    hasTitle: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Whether to show the title in the toast'
    },
    hasDescription: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Whether to show the description in the toast'
    },
    customColor: {
      type: 'string',
      required: false,
      default: 'undefined',
      description: 'Custom background color override (hex, rgb, etc.)'
    },
    autoDismissDelay: {
      type: 'number',
      required: false,
      default: '5000',
      description: 'Delay in milliseconds before auto-closing'
    },
    hasAutoDismiss: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Whether the toast automatically closes after a delay'
    }
  },

  // ToastProvider Props
  providerProps: {
    defaultPosition: {
      type: "'top-left' | 'top-center' | 'top-right' | 'bottom-left' | 'bottom-center' | 'bottom-right'",
      required: false,
      default: "'top-right'",
      description: 'Default position for all toasts'
    },
    defaultSize: {
      type: "'xs' | 'sm' | 'md' | 'lg' | 'xl'",
      required: false,
      default: "'md'",
      description: 'Default size for all toasts'
    },
    autoDismissDelay: {
      type: 'number',
      required: false,
      default: '5000',
      description: 'Default auto-dismiss delay for all toasts (milliseconds)'
    },
    hasAutoDismiss: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Default auto-dismiss behavior for all toasts'
    },
    hasIcon: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Default icon visibility for all toasts'
    },
    hasCloseButton: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Default close button visibility for all toasts'
    },
    hasTitle: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Default title visibility for all toasts'
    },
    hasDescription: {
      type: 'boolean',
      required: false,
      default: 'true',
      description: 'Default description visibility for all toasts'
    },
    customColor: {
      type: 'string',
      required: false,
      default: 'undefined',
      description: 'Custom background color override for all toasts'
    }
  },

  // useToast Hook
  useToast: {
    description: 'Hook for managing toast notifications and positioning',
    returns: {
      showToast: {
        type: 'function',
        description: 'Function to show a new toast',
        parameters: {
          options: {
            type: 'object',
            description: 'Toast options',
            properties: {
              title: 'string (required) - Toast title',
              description: 'string (optional) - Toast description',
              variant: "'success' | 'error' | 'info' | 'warning' | 'danger' (required) - Toast variant",
              size: "'xs' | 'sm' | 'md' | 'lg' | 'xl' (optional) - Toast size"
            }
          }
        },
        example: `const { showToast } = useToast();

showToast({
  title: 'Success!',
  description: 'Your action was completed successfully.',
  variant: 'success',
  size: 'md'
});`
      },
      updatePosition: {
        type: 'function',
        description: 'Function to update the position of all toasts',
        parameters: {
          position: {
            type: "'top-left' | 'top-center' | 'top-right' | 'bottom-left' | 'bottom-center' | 'bottom-right'",
            description: 'New position for toasts'
          }
        },
        example: `const { updatePosition } = useToast();

// Change position dynamically
updatePosition('bottom-center');`
      }
    }
  },

  // Usage Examples
  examples: {
    basic: {
      title: 'Basic Toast Usage',
      description: 'Simple toast notification with default settings',
      code: `import { ToastProvider, useToast } from '@acutrack-bookprint/acutrack-ds';

function App() {
  return (
    <ToastProvider>
      <MyComponent />
    </ToastProvider>
  );
}

function MyComponent() {
  const { showToast } = useToast();

  const handleSuccess = () => {
    showToast({
      title: 'Success!',
      description: 'Your action was completed successfully.',
      variant: 'success'
    });
  };

  return (
    <button onClick={handleSuccess}>
      Show Success Toast
    </button>
  );
}`
    },
    customPosition: {
      title: 'Custom Position',
      description: 'Toast with custom positioning',
      code: `import { ToastProvider, useToast } from '@acutrack-bookprint/acutrack-ds';

function App() {
  return (
    <ToastProvider defaultPosition="bottom-center">
      <MyComponent />
    </ToastProvider>
  );
}`
    },
    customSize: {
      title: 'Custom Size',
      description: 'Toast with custom size',
      code: `const { showToast } = useToast();

showToast({
  title: 'Large Message',
  description: 'This is a large toast notification.',
  variant: 'info'
});`
    },
    noAutoClose: {
      title: 'No Auto Close',
      description: 'Toast that requires manual dismissal',
      code: `const { showToast } = useToast();

showToast({
  title: 'Important Message',
  description: 'This toast will not auto-close.',
  variant: 'warning'
});`
    },
    withActions: {
      title: 'Toast with Actions',
      description: 'Toast with action buttons',
      code: `const { showToast } = useToast();

showToast({
  title: 'File Uploaded',
  description: 'Your file has been uploaded successfully.',
  variant: 'success',
  actions: [
    {
      label: 'View',
      onClick: () => console.log('View file')
    },
    {
      label: 'Download',
      onClick: () => console.log('Download file')
    }
  ]
});`
    }
  },

  // Best Practices
  bestPractices: {
    do: [
      'Use appropriate variants for different message types',
      'Keep toast messages concise and clear',
      'Use success toasts for completed actions',
      'Use error toasts for failed operations',
      'Use warning toasts for important information',
      'Use info toasts for general updates',
      'Position toasts where they won\'t interfere with user actions',
      'Provide clear, actionable descriptions when possible',
      'Use consistent timing for auto-close delays',
      'Test toast behavior in both light and dark modes'
    ],
    dont: [
      'Don\'t use toasts for critical errors that require immediate attention',
      'Don\'t stack too many toasts simultaneously',
      'Don\'t use toasts for information that should be persistent',
      'Don\'t make toast messages too long or complex',
      'Don\'t use toasts for form validation errors',
      'Don\'t override toast positioning without good reason',
      'Don\'t use toasts for navigation or page changes',
      'Don\'t make toasts auto-close too quickly for important messages',
      'Don\'t use toasts as the primary way to display errors',
      'Don\'t ignore accessibility considerations'
    ]
  },

  // Accessibility
  accessibility: {
    features: [
      'Proper ARIA labels and roles',
      'Keyboard navigation support',
      'Screen reader compatibility',
      'Focus management',
      'High contrast support',
      'Reduced motion support'
    ],
    guidelines: [
      'Ensure toast messages are announced by screen readers',
      'Provide alternative ways to access toast content',
      'Use semantic HTML elements',
      'Maintain proper focus order',
      'Test with keyboard-only navigation',
      'Ensure sufficient color contrast'
    ]
  },

  // Design System Integration
  designSystem: {
    colors: [
      'Uses design system color tokens',
      'Supports light and dark themes',
      'Consistent with other component colors',
      'Proper contrast ratios maintained'
    ],
    typography: [
      'Uses design system font sizes',
      'Consistent line heights and spacing',
      'Proper text hierarchy'
    ],
    spacing: [
      'Uses design system spacing tokens',
      'Consistent padding and margins',
      'Proper component spacing'
    ],
    icons: [
      'Uses Material Symbols icons',
      'Consistent icon sizing',
      'Proper icon alignment'
    ]
  },

  // Common Issues and Solutions
  troubleshooting: {
    'Toast not appearing': {
      cause: 'ToastProvider not wrapping the component',
      solution: 'Ensure ToastProvider wraps your app or component tree'
    },
    'Toast appearing in wrong position': {
      cause: 'Position prop not set correctly',
      solution: 'Check the position prop value and ToastProvider defaultPosition'
    },
    'Toast not auto-closing': {
      cause: 'autoClose prop set to false',
      solution: 'Set autoClose to true or check autoCloseDelay value'
    },
    'Multiple toasts overlapping': {
      cause: 'Too many toasts or incorrect z-index',
      solution: 'Limit maxToasts or adjust z-index in ToastProvider'
    },
    'Toast styling not applying': {
      cause: 'CSS not imported or design system colors not available',
      solution: 'Import the design system CSS and ensure proper color tokens'
    }
  },

  // Related Components
  relatedComponents: [
    'Alert - For persistent notifications',
    'Modal - For important dialogs',
    'Popover - For contextual information',
    'Tooltip - For brief help text',
    'Banner - For page-level notifications'
  ],

  // Changelog
  changelog: [
    {
      version: '1.0.68',
      date: '2024-12-19',
      changes: [
        'Fixed position communication between ToastTestPage and ToastProvider',
        'Resolved toast positioning issues for top-left and top-right positions',
        'Fixed center positioning for both top-center and bottom-center positions',
        'Implemented comprehensive flexbox-based positioning system',
        'Added dynamic toast wrapper classes for different position types',
        'Updated Tailwind safelist to include w-auto class for proper positioning',
        'Created comprehensive ToastTestPage with interactive playground',
        'Added real-time position updates through useToast hook integration',
        'Implemented code generation feature for developers',
        'Added tabbed interface with Preview and Code sections',
        'Enhanced debug logging for position updates',
        'Improved toast container positioning with proper width constraints'
      ]
    },
    {
      version: '1.0.66',
      date: '2024-12-19',
      changes: [
        'Simplified Toast component architecture',
        'Removed over 100 unused props for better performance',
        'Fixed dropdown clipping issues in test pages',
        'Improved z-index management for proper layering',
        'Enhanced test page with comprehensive configuration',
        'Better positioning system with proper Tailwind classes',
        'Improved accessibility and screen reader support',
        'Optimized bundle size and performance'
      ]
    },
    {
      version: '1.0.0',
      date: '2024-01-01',
      changes: [
        'Initial release of Toast component',
        'Support for all variants (success, error, warning, info, danger)',
        'Customizable positioning and sizing',
        'Auto-close functionality',
        'Dark mode support',
        'Accessibility features',
        'Design system integration'
      ]
    }
  ]
};

export default ToastDocumentation;
